First, for operating system, Linux or MacOS is recommended. If you use Windows, please change all the directories leading with "/tmp" in the file "spComparison/spcmp.py" and this "demo.ipynb" file into any directory you prefer (e.g. "C:\tmp") in your Windows system. In addition, for Windows user, it is recommended to install Cygwin (https://www.cygwin.com/) first, in order to have a Linux-like environment with tools like Bash, bc, etc.

Second, you need to have a functional Python 3.6 or 3.7 installed. Installing Python through MiniConda (https://docs.conda.io/en/latest/miniconda.html) is recommended. Then use pip to install NumPy and Numba.

Third, you need to have a functional GMT5 installed. Please refer to http://gmt.soest.hawaii.edu/projects/gmt/wiki/Installing for its installing instructions.

# Introducing Use of the Core Function "spa_angpre_len_dif"

Then, let's get started with an example from pair A (Figure 3a in the paper https://github.com/f-i/APWP_similarity).

In [2]:
## import Python modules
# import Python's sys (System-specific parameters and functions) module
import sys
# import Python's copytree module
from distutils.dir_util import copy_tree
# add this directory for Python searching non-standard modules, like our own modules
sys.path.append('.')  #only accept relative dir
# import my own module
import spComparison.spcmp as cmp

## pair A (Figure 3a)

In [ ]:
## calculate pair A's difference scores in terms of space (coeval poles), angle and
## length (coeval segments); NOTE THAT THIS COULD TAKE HOURS (Depends on your OS and Hardwares)
# copy raw VGP locations (only for poles with their N greater than 25) to /tmp
copy_tree("data/Laurentia_RM", "/tmp/Laurentia_RM")
copy_tree("data/rot_version/Laurentia_RM_1deg","/tmp/Laurentia_RM_1deg")
# rum our spa_angpre_len_dif function to calculate difference scores
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_1deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairA_LaurentiaRM_LaurentiaRM1deg.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	0.06950468110321331	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	0.07012239182561895	0	nan	nan	0.	0	273.743775125	273.149982677	0.5238885648671224	0.5238885648601654
2	20.0	0.12965232767905435	0	0.	0	0.	0	139.00062497750002	139.0006249778	4.4390170905897985	4.439017090586511


The above results displayed have 12 columns.

	00_no:			index / order number
	01_tstop:		pole age
	10_spa_pol_dif:	spacial difference of each coeval pole pair
	11_spa_pol_tes:	1 or 0, i.e., distinguishable or indistinguishable
					spacial relationship of coeval uncertainties
	20_ang_seg_dif:	angular difference of each coeval segment pair
	21_ang_seg_tes:	1 or 0, i.e., a significant or no significant angular difference,
					based on random sampling from pole uncertainties
	30_len_seg_dif:	length difference of each coeval segment pair
	31_len_seg_tes:	1 or 0, i.e., a significant or no significant length difference,
					based on random sampling from pole uncertainties
    22_course_seg1: trajectory I's each segment's orientational change
    23_course_seg2: trajectory II's each contemporary segment's orientational change
    32_len_seg1: trajectory I's each segment's length
    33_len_seg2: trajectory II's each contemporary segment's length

In [2]:
## reproduce results for pair A in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairA=cmp.ppf3('data/0.result_tables/pairA_LaurentiaRM_LaurentiaRM1deg.txt')
#pairA_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairA,'0',0,100)
cmp.calc_nt(pairA,'1',100,200)
cmp.calc_nt(pairA,'2',200,300)
cmp.calc_nt(pairA,'3',300,400)
cmp.calc_nt(pairA,'4',400,530)
cmp.calc_nt(pairA,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.00402313	0	0	0	0	100	0.00134104
0.00798601	0	0	1	100	200	0.002662
0.0126793	0	0	2	200	300	0.00422643
0.0182357	0	0	3	300	400	0.00607858
0.0194666	0	0	4	400	530	0.00648888
0.0128715	0	0	5	0	530	0.00429051


The above results displayed own 6 columns.

    ds: untested mean spatial difference score
    da: untested mean angular difference score
    dl: untested mean length difference score
    no: index / order number
    t0: beginning pole age of the specific path
    t1: end pole age of the specific path
    equalWeighCPD: CPD from equally weighted metrics

In [3]:
## reproduce results for pair A in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairA.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairA,'0',0,100)
cmp.calc(pairA,'1',100,200)
cmp.calc(pairA,'2',200,300)
cmp.calc(pairA,'3',300,400)
cmp.calc(pairA,'4',400,530)
cmp.calc(pairA,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0	0	0	0	0	100	0
0	0	0	1	100	200	0
0	0	0	2	200	300	0
0.0068817	0	0	3	300	400	0.0022939
0.00413091	0	0	4	400	530	0.00137697
0.00247281	0	0	5	0	530	0.000824268


The above results displayed also own 6 columns.

    ds: mean significant spatial difference score
    da: mean significant angular difference score
    dl: mean significant length difference score
    no: index / order number
    t0: beginning pole age of the specific path
    t1: end pole age of the specific path
    equalWeighCPD: CPD from equally weighted metrics

# Reproducing Results for the Other Five Pairs Using the Same Procedure Above

## pair B

In [ ]:
## calculate pair B's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
copy_tree("data/rot_version/Laurentia_RM_15deg","/tmp/Laurentia_RM_15deg")
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_15deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairB_LaurentiaRM_LaurentiaRM15deg.txt

In [2]:
## reproduce results for pair B in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairB=cmp.ppf3('data/0.result_tables/pairB_LaurentiaRM_LaurentiaRM15deg.txt')
#pairB_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairB,'0',0,100)
cmp.calc_nt(pairB,'1',100,200)
cmp.calc_nt(pairB,'2',200,300)
cmp.calc_nt(pairB,'3',300,400)
cmp.calc_nt(pairB,'4',400,530)
cmp.calc_nt(pairB,'5',0,530)

ds	da	dl	no	t0	t1
0.06018565442887694	0.0	0.0	0	0	100
0.11951075417785047	0.0	0.0	1	100	200
0.18987624618932064	0.0	0.0	2	200	300
0.27341452084170853	0.0	0.0	3	300	400
0.29195711704472566	0.0	0.0	4	400	530
0.1928983752638573	0.0	0.0	5	0	530


In [3]:
## reproduce results for pair B in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairB.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairB,'0',0,100)
cmp.calc(pairB,'1',100,200)
cmp.calc(pairB,'2',200,300)
cmp.calc(pairB,'3',300,400)
cmp.calc(pairB,'4',400,530)
cmp.calc(pairB,'5',0,530)

ds	da	dl	no	t0	t1
0.0	0.0	0.0	0	0	100
0.02452207584754257	0.0	0.0	1	100	200
0.17214017750222857	0.0	0.0	2	200	300
0.27341452084170853	0.0	0.0	3	300	400
0.18753232193044467	0.0	0.0	4	400	530
0.13206089460320947	0.0	0.0	5	0	530


## pair C

In [ ]:
## calculate pair C's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
copy_tree("data/rot_version/Laurentia_RM_po_15deg","/tmp/Laurentia_RM_po_15deg")
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_po_15deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairC_LaurentiaRM_LaurentiaRMpo15deg.txt

In [4]:
## reproduce results for pair C in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairC=cmp.ppf3('data/0.result_tables/pairC_LaurentiaRM_LaurentiaRMpo15deg.txt')
#pairC_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairC,'0',0,100)
cmp.calc_nt(pairC,'1',100,200)
cmp.calc_nt(pairC,'2',200,300)
cmp.calc_nt(pairC,'3',300,400)
cmp.calc_nt(pairC,'4',400,530)
cmp.calc_nt(pairC,'5',0,530)

ds	da	dl	no	t0	t1
0.060179524592502295	0.24790732639537033	1.0905640046804856e-05	0	0	100
0.11948928991824084	0.2283843537798148	0.0001096116344275399	1	100	200
0.1897653823765637	0.4897948633596197	2.7047084440588792e-05	2	200	300
0.27310680364795525	0.3579541481306791	0.00011034440982000611	3	300	400
0.2918073308934256	0.3278328522055788	0.0001996424474577322	4	400	530
0.19278564167708362	0.32821897320673443	9.763093404339463e-05	5	0	530


In [5]:
## reproduce results for pair C in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairC.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairC,'0',0,100)
cmp.calc(pairC,'1',100,200)
cmp.calc(pairC,'2',200,300)
cmp.calc(pairC,'3',300,400)
cmp.calc(pairC,'4',400,530)
cmp.calc(pairC,'5',0,530)

ds	da	dl	no	t0	t1
0.0	0.0	0.0	0	0	100
0.024511167393754963	0.0	0.0	1	100	200
0.17204153054667518	0.10183802030481481	0.0	2	200	300
0.27310680364795525	0.0983773321814815	2.2421740928911606e-05	3	300	400
0.18740701021938655	0.0	7.216273757176646e-07	4	400	530
0.13196018041953333	0.03465265716108974	4.407520097612184e-06	5	0	530


## pair D

In [ ]:
## calculate pair D's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
copy_tree("data/rot_version/Laurentia_RM_p_15deg","/tmp/Laurentia_RM_p_15deg")
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_p_15deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairD_LaurentiaRM_LaurentiaRMp15deg.txt

In [6]:
## reproduce results for pair D in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairD=cmp.ppf3('data/0.result_tables/pairD_LaurentiaRM_LaurentiaRMp15deg.txt')
#pairD_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairD,'0',0,100)
cmp.calc_nt(pairD,'1',100,200)
cmp.calc_nt(pairD,'2',200,300)
cmp.calc_nt(pairD,'3',300,400)
cmp.calc_nt(pairD,'4',400,530)
cmp.calc_nt(pairD,'5',0,530)

ds	da	dl	no	t0	t1
0.0601804953535425	0.5020896657221604	0.1242469695162171	0	0	100
0.11946892740034999	0.5922044939625246	0.1329337877885109	1	100	200
0.18976154831555006	0.5804934235719901	0.36609065818561376	2	200	300
0.2731263641477164	0.5121743766190309	0.5162692905782692	3	300	400
0.2916446594852434	0.5685291350304167	0.4509314758331272	4	400	530
0.19273680028530343	0.5457373845189663	0.32561351408522193	5	0	530


In [7]:
## reproduce results for pair D in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairD.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairD,'0',0,100)
cmp.calc(pairD,'1',100,200)
cmp.calc(pairD,'2',200,300)
cmp.calc(pairD,'3',300,400)
cmp.calc(pairD,'4',400,530)
cmp.calc(pairD,'5',0,530)

ds	da	dl	no	t0	t1
0.0	0.0	0.0	0	0	100
0.0245146198623683	0.0	0.0	1	100	200
0.1720324242622825	0.09753166968487655	0.16947427972050066	2	200	300
0.2731263641477164	0.20273891366180868	0.3774520251613964	3	300	400
0.18735175977842114	0.12854420865305555	0.2501510071082986	4	400	530
0.13194281599422122	0.08163395680686218	0.16455143662692176	5	0	530


## pair E

In [ ]:
## calculate pair E's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
copy_tree("data/rot_version/Laurentia_RM_45deg","/tmp/Laurentia_RM_45deg")
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_45deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairE_LaurentiaRM_LaurentiaRM45deg.txt

In [8]:
## reproduce results for pair E in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairE=cmp.ppf3('data/0.result_tables/pairE_LaurentiaRM_LaurentiaRM45deg.txt')
#pairE_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairE,'0',0,100)
cmp.calc_nt(pairE,'1',100,200)
cmp.calc_nt(pairE,'2',200,300)
cmp.calc_nt(pairE,'3',300,400)
cmp.calc_nt(pairE,'4',400,530)
cmp.calc_nt(pairE,'5',0,530)

ds	da	dl	no	t0	t1
0.176682672802216	0.0	0.0	0	0	100
0.35176917694730764	0.0	0.0	1	100	200
0.561929737862065	0.0	0.0	2	200	300
0.8171727224060709	0.0	0.0	3	300	400
0.8747843279351402	0.0	0.0	4	400	530
0.5744135257153348	0.0	0.0	5	0	530


In [9]:
## reproduce results for pair E in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairE.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairE,'0',0,100)
cmp.calc(pairE,'1',100,200)
cmp.calc(pairE,'2',200,300)
cmp.calc(pairE,'3',300,400)
cmp.calc(pairE,'4',400,530)
cmp.calc(pairE,'5',0,530)

ds	da	dl	no	t0	t1
0.13418921255437907	0.0	0.0	0	0	100
0.3307515913351165	0.0	0.0	1	100	200
0.561929737862065	0.0	0.0	2	200	300
0.8171727224060709	0.0	0.0	3	300	400
0.8747843279351402	0.0	0.0	4	400	530
0.5657574504796643	0.0	0.0	5	0	530


## pair F

In [ ]:
## calculate pair F's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
copy_tree("data/Laurentia_RM_NoErr","/tmp/Laurentia_RM_NoErr")
copy_tree("data/rot_version/Laurentia_RM_NoErr_45deg","/tmp/Laurentia_RM_NoErr_45deg")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoErr.txt',
                       'data/rot_version/Laurentia_RM_NoErr_45deg.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairF_LaurentiaRMnoErr_LaurentiaRMnoErr45deg.txt

In [10]:
## reproduce results for pair F in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairF=cmp.ppf3('data/0.result_tables/pairF_LaurentiaRMnoErr_LaurentiaRMnoErr45deg.txt')
#pairF_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairF,'0',0,100)
cmp.calc_nt(pairF,'1',100,200)
cmp.calc_nt(pairF,'2',200,300)
cmp.calc_nt(pairF,'3',300,400)
cmp.calc_nt(pairF,'4',400,530)
cmp.calc_nt(pairF,'5',0,530)

ds	da	dl	no	t0	t1
0.176682672802216	0.0	0.0	0	0	100
0.35176917694730764	0.0	0.0	1	100	200
0.561929737862065	0.0	0.0	2	200	300
0.8171727224060709	0.0	0.0	3	300	400
0.8747843279351402	0.0	0.0	4	400	530
0.5744135257153348	0.0	0.0	5	0	530


In [11]:
## reproduce results for pair F in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairF.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairF,'0',0,100)
cmp.calc(pairF,'1',100,200)
cmp.calc(pairF,'2',200,300)
cmp.calc(pairF,'3',300,400)
cmp.calc(pairF,'4',400,530)
cmp.calc(pairF,'5',0,530)

ds	da	dl	no	t0	t1
0.176682672802216	0.0	0.0	0	0	100
0.35176917694730764	0.0	0.0	1	100	200
0.561929737862065	0.0	0.0	2	200	300
0.8171727224060709	0.0	0.0	3	300	400
0.8747843279351402	0.0	0.0	4	400	530
0.5744135257153348	0.0	0.0	5	0	530


## Figure 10 pair (a)

In [2]:
## calculate pair Baltica and Laurentia's difference scores in terms of space (coeval poles),
## length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurentia_RM_GMAP2015BALcoord","/tmp/Laurentia_RM_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_RM_NoIntp","/tmp/Baltica_StableEurope_RM_NoIntp")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_RM_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_RM_NoIntp.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LaureRMgmap2015BALcoord_BalRM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	5.147335987512224	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	7.001796297426207	1	nan	nan	0.8161605555610272	0	285.020779667	77.8897393654	1.3616050844405654	2.1777656400015926
2	20.0	6.180885121529476	0	170.87375047330002	0	0.8169389443726365	0	123.2517817197	-47.621968753600015	4.066159402442237	3.2492204580696002
3	30.0	3.629579311648592	0	107.51717069659999	0	0.10955289997333595	0	55.60199635740001	163.119167054	1.5999677253451032	1.7095206253184392
4	40.0	6.05879741900248	1	44.17952569899998	0	3.2576549669748127	0	-107.147267209	-62.967741510000025	1.5393184649983633	4.796973431973176
5	50.0	1.2718160782072718	0	150.98197492409997	0	1.5236538880764936	0	73.7897937579	-135.22823131800004	5.534146234663954	4.01049234658746
6	60.0	1.8415932399735546	0	92.74538163900007	0	0.18825885282851396	0	-55.18113608400003	37.5642455550

array([( 0,   0.,  5.14733599, 0,          nan, nan,         nan, nan,           nan,             nan,         nan,         nan),
       ( 1,  10.,  7.0017963 , 1,          nan, nan,  0.81616056,  0.,  285.02077967,  7.78897394e+01,  1.36160508,  2.17776564),
       ( 2,  20.,  6.18088512, 0, 170.87375047,  0.,  0.81693894,  0.,  123.25178172, -4.76219688e+01,  4.0661594 ,  3.24922046),
       ( 3,  30.,  3.62957931, 0, 107.5171707 ,  0.,  0.1095529 ,  0.,   55.60199636,  1.63119167e+02,  1.59996773,  1.70952063),
       ( 4,  40.,  6.05879742, 1,  44.1795257 ,  0.,  3.25765497,  0., -107.14726721, -6.29677415e+01,  1.53931846,  4.79697343),
       ( 5,  50.,  1.27181608, 0, 150.98197492,  0.,  1.52365389,  0.,   73.78979376, -1.35228231e+02,  5.53414623,  4.01049235),
       ( 6,  60.,  1.84159324, 0,  92.74538164,  0.,  0.18825885,  0.,  -55.18113608,  3.75642456e+01,  1.01983631,  0.83157746),
       ( 7,  70.,  4.27931258, 0, 168.12385383,  0.,  1.52502453,  0., -172.99840294, -4.8

In [6]:
## reproduce d_notest results for the APWP pair in Figure 10(a) using the output above
## from the function cmp.spa_angpre_len_dif
pair_10a=cmp.ppf3('data/0.result_tables/LaureRMgmap2015BALcoord_BalRM.txt')
#10a_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10a,'0',0,100)
cmp.calc_nt(pair_10a,'1',100,200)
cmp.calc_nt(pair_10a,'2',200,280)
cmp.calc_nt(pair_10a,'3',280,320)
cmp.calc_nt(pair_10a,'4',320,430)
cmp.calc_nt(pair_10a,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.103919	0.623628	0.0417716	0	0	100	0.256439
0.156899	0.227443	0.126521	1	100	200	0.170288
0.123066	0.365675	0.133068	2	200	280	0.20727
0.307399	0.521818	0.0877977	3	280	320	0.305671
0.151982	0.221145	0.142501	4	320	430	0.171876
0.151657	0.376146	0.108516	5	0	430	0.212106


In [7]:
## significant results
#10a_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10a,'0',0,100)
cmp.calc(pair_10a,'1',100,200)
cmp.calc(pair_10a,'2',200,280)
cmp.calc(pair_10a,'3',280,320)
cmp.calc(pair_10a,'4',320,430)
cmp.calc(pair_10a,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0609611	0	0	0	0	100	0.0203204
0.103744	0	0	1	100	200	0.0345812
0.0917036	0	0	2	200	280	0.0305679
0.307399	0	0	3	280	320	0.102466
0.0474133	0	0	4	320	430	0.0158044
0.100182	0	0	5	0	430	0.0333941


## Figure 10 pair (b)

In [8]:
## calculate pair Baltica and Laurentia (I-error corrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord","/tmp/Laurentia_F6_RM_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_F6_RM_NoIntp","/tmp/Baltica_StableEurope_F6_RM_NoIntp")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_NoIntp.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LaureF6RMgmap2015BALcoord_BalF6RM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	5.511605279531963	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	6.838840981689385	1	nan	nan	1.4007267098175318	0	223.662074975	77.8897393668	0.7770389303206573	2.177765640138189
2	20.0	6.131170778282259	0	164.7897193518	0	0.6779218946133092	0	147.5883118936	-47.621968754600005	3.9271423527523797	3.2492204581390705
3	30.0	3.6295793119117548	0	153.23538056100003	0	0.8706639618492285	0	9.883786493999992	163.11916705500002	0.8388566638504635	1.709520625699692
4	40.0	5.24200286568283	1	44.19072628300003	0	3.9727608013639077	0	-18.777015226999993	-62.967741510000025	0.8242126305540449	4.7969734319179524
5	50.0	1.5761918278308356	0	148.86528379919997	0	2.4815400896795192	0	53.11391558979999	-158.02080061100003	5.747157815932648	3.265617726253129
6	60.0	0.8449292600628675	0	125.61853399120999	0	0.9144896783851173	0	-59.276632990999985	6

array([( 0,   0.,  5.51160528, 1,          nan, nan,         nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  6.83884098, 1,          nan, nan,  1.40072671,  0.,  223.66207497,   77.88973937,  0.77703893,  2.17776564),
       ( 2,  20.,  6.13117078, 0, 164.78971935,  0.,  0.67792189,  0.,  147.58831189,  -47.62196875,  3.92714235,  3.24922046),
       ( 3,  30.,  3.62957931, 0, 153.23538056,  0.,  0.87066396,  0.,    9.88378649,  163.11916706,  0.83885666,  1.70952063),
       ( 4,  40.,  5.24200287, 1,  44.19072628,  0.,  3.9727608 ,  0.,  -18.77701523,  -62.96774151,  0.82421263,  4.79697343),
       ( 5,  50.,  1.57619183, 0, 148.8652838 ,  0.,  2.48154009,  0.,   53.11391559, -158.02080061,  5.74715782,  3.26561773),
       ( 6,  60.,  0.84492926, 0, 125.61853399,  0.,  0.91448968,  0.,  -59.27663299,   66.341901  ,  0.52353728,  1.43802695),
       ( 7,  70.,  6.07027007, 0, 170.45138299,  0.,  1.04189461,  0., -173.87000113,   -3.41861814,  3.

In [9]:
## reproduce d_notest results for pair (b) in Figure 10(b) using the output above
## from the function cmp.spa_angpre_len_dif
pair_10b=cmp.ppf3('data/0.result_tables/LaureF6RMgmap2015BALcoord_BalF6RM.txt')
#10b_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10b,'0',0,100)
cmp.calc_nt(pair_10b,'1',100,200)
cmp.calc_nt(pair_10b,'2',200,280)
cmp.calc_nt(pair_10b,'3',280,320)
cmp.calc_nt(pair_10b,'4',320,430)
cmp.calc_nt(pair_10b,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.107992	0.583144	0.0603922	0	0	100	0.250509
0.145741	0.219977	0.130902	1	100	200	0.16554
0.106125	0.365272	0.145239	2	200	280	0.205545
0.289963	0.565152	0.109864	3	280	320	0.32166
0.144176	0.164366	0.142085	4	320	430	0.150209
0.141983	0.361833	0.118076	5	0	430	0.207297


In [10]:
## significant results
#10b_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10b,'0',0,100)
cmp.calc(pair_10b,'1',100,200)
cmp.calc(pair_10b,'2',200,280)
cmp.calc(pair_10b,'3',280,320)
cmp.calc(pair_10b,'4',320,430)
cmp.calc(pair_10b,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0724854	0	0	0	0	100	0.0241618
0.0452449	0	0	1	100	200	0.0150816
0.0265822	0	0	2	200	280	0.00886074
0.289963	0	0	3	280	320	0.0966544
0.0503174	0	0.0665741	4	320	430	0.0389638
0.0717395	0	0.0170306	5	0	430	0.02959


## Figure 10 pair (a)'s Baltica (Uncorrected) vs pair (b)'s Laurentia (Corrected)

In [3]:
## calculate pair Baltica (Uncorrected) and Laurentia (Corrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord","/tmp/Laurentia_F6_RM_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_RM_NoIntp","/tmp/Baltica_StableEurope_RM_NoIntp")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_RM_NoIntp.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LaureF6RMgmap2015BALcoord_BalRM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	5.5116052791749235	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	6.838840981328586	1	nan	nan	1.4007267096809353	0	223.662074975	77.8897393654	0.7770389303206573	2.1777656400015926
2	20.0	6.131170777853502	0	164.78971935279998	0	0.6779218946827794	0	147.5883118936	-47.621968753600015	3.9271423527523797	3.2492204580696002
3	30.0	3.6295793118554847	0	153.23538056	0	0.8706639614679756	0	9.883786493999992	163.119167054	0.8388566638504635	1.7095206253184392
4	40.0	5.24200286568283	1	44.19072628300003	0	3.972760801419131	0	-18.777015226999993	-62.967741510000025	0.8242126305540449	4.796973431973176
5	50.0	1.00981254598451	0	171.65785309219996	0	1.736665469345188	0	53.11391558979999	-135.22823131800004	5.747157815932648	4.01049234658746
6	60.0	1.1105263534761594	0	96.84087854600003	0	0.30804018433407554	0	-59.276632990999985	37.56424555

array([( 0,   0.,  5.51160528, 1,          nan, nan,         nan, nan,           nan,             nan,         nan,         nan),
       ( 1,  10.,  6.83884098, 1,          nan, nan,  1.40072671,  0.,  223.66207497,  7.78897394e+01,  0.77703893,  2.17776564),
       ( 2,  20.,  6.13117078, 0, 164.78971935,  0.,  0.67792189,  0.,  147.58831189, -4.76219688e+01,  3.92714235,  3.24922046),
       ( 3,  30.,  3.62957931, 0, 153.23538056,  0.,  0.87066396,  0.,    9.88378649,  1.63119167e+02,  0.83885666,  1.70952063),
       ( 4,  40.,  5.24200287, 1,  44.19072628,  0.,  3.9727608 ,  0.,  -18.77701523, -6.29677415e+01,  0.82421263,  4.79697343),
       ( 5,  50.,  1.00981255, 0, 171.65785309,  0.,  1.73666547,  0.,   53.11391559, -1.35228231e+02,  5.74715782,  4.01049235),
       ( 6,  60.,  1.11052635, 0,  96.84087855,  0.,  0.30804018,  0.,  -59.27663299,  3.75642456e+01,  0.52353728,  0.83157746),
       ( 7,  70.,  6.07027007, 0, 168.99545202,  0.,  2.00348793,  0., -173.87000113, -4.8

In [7]:
pair_10aB10bL=cmp.ppf3('data/0.result_tables/LaureF6RMgmap2015BALcoord_BalRM.txt')
#10aB10bL_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10aB10bL,'0',0,100)
cmp.calc_nt(pair_10aB10bL,'1',100,200)
cmp.calc_nt(pair_10aB10bL,'2',200,280)
cmp.calc_nt(pair_10aB10bL,'3',280,320)
cmp.calc_nt(pair_10aB10bL,'4',320,430)
cmp.calc_nt(pair_10aB10bL,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.107445	0.583176	0.0589476	0	0	100	0.249856
0.143976	0.227608	0.135968	1	100	200	0.169184
0.108438	0.388919	0.13473	2	200	280	0.210696
0.295427	0.545904	0.106805	3	280	320	0.316045
0.145257	0.174735	0.142433	4	320	430	0.154142
0.143217	0.367449	0.116767	5	0	430	0.209144


In [8]:
## significant results
#10aB10bL_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10aB10bL,'0',0,100)
cmp.calc(pair_10aB10bL,'1',100,200)
cmp.calc(pair_10aB10bL,'2',200,280)
cmp.calc(pair_10aB10bL,'3',280,320)
cmp.calc(pair_10aB10bL,'4',320,430)
cmp.calc(pair_10aB10bL,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0724854	0	0	0	0	100	0.0241618
0.0625872	0	0	1	100	200	0.0208624
0.0270311	0	0	2	200	280	0.00901037
0.295427	0	0	3	280	320	0.0984758
0.0503174	0	0.0665741	4	320	430	0.0389638
0.0766962	0	0.0170306	5	0	430	0.0312423


## Figure 10 pair (a)'s Laurentia (Uncorrected) vs pair (b)'s Baltica (Corrected)

In [4]:
## calculate pair Baltica (Corrected) and Laurentia (Uncorrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurentia_RM_GMAP2015BALcoord","/tmp/Laurentia_RM_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_F6_RM_NoIntp","/tmp/Baltica_StableEurope_F6_RM_NoIntp")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_RM_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_NoIntp.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LaureRMgmap2015BALcoord_BalF6RM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	5.147335987865811	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	7.001796297776876	1	nan	nan	0.8161605556976237	0	285.020779667	77.8897393668	1.3616050844405654	2.177765640138189
2	20.0	6.180885121950783	0	170.8737504743	0	0.8169389443031663	0	123.2517817197	-47.621968754600005	4.066159402442237	3.2492204581390705
3	30.0	3.6295793117047612	0	107.51717069760001	0	0.10955290035458876	0	55.60199635740001	163.11916705500002	1.5999677253451032	1.709520625699692
4	40.0	6.05879741900248	1	44.17952569899998	0	3.2576549669195893	0	-107.147267209	-62.967741510000025	1.5393184649983633	4.7969734319179524
5	50.0	1.7415656577726388	0	128.1894056311	0	2.268528508410825	0	73.7897937579	-158.02080061100003	5.534146234663954	3.265617726253129
6	60.0	1.844676598606582	0	121.52303708421003	0	0.4181906412225278	0	-55.18113608400003	66.34190100021	1.

array([( 0,   0.,  5.14733599, 1,          nan, nan,         nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  7.0017963 , 1,          nan, nan,  0.81616056,  0.,  285.02077967,   77.88973937,  1.36160508,  2.17776564),
       ( 2,  20.,  6.18088512, 0, 170.87375047,  0.,  0.81693894,  0.,  123.25178172,  -47.62196875,  4.0661594 ,  3.24922046),
       ( 3,  30.,  3.62957931, 0, 107.5171707 ,  0.,  0.1095529 ,  0.,   55.60199636,  163.11916706,  1.59996773,  1.70952063),
       ( 4,  40.,  6.05879742, 1,  44.1795257 ,  0.,  3.25765497,  0., -107.14726721,  -62.96774151,  1.53931846,  4.79697343),
       ( 5,  50.,  1.74156566, 0, 128.18940563,  0.,  2.26852851,  0.,   73.78979376, -158.02080061,  5.53414623,  3.26561773),
       ( 6,  60.,  1.8446766 , 0, 121.52303708,  0.,  0.41819064,  0.,  -55.18113608,   66.341901  ,  1.01983631,  1.43802695),
       ( 7,  70.,  4.27931258, 0, 169.5797848 ,  0.,  0.56343122,  0., -172.99840294,   -3.41861814,  3.

In [9]:
pair_10aL10bB=cmp.ppf3('data/0.result_tables/LaureRMgmap2015BALcoord_BalF6RM.txt')
#10aL10bB_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10aL10bB,'0',0,100)
cmp.calc_nt(pair_10aL10bB,'1',100,200)
cmp.calc_nt(pair_10aL10bB,'2',200,280)
cmp.calc_nt(pair_10aL10bB,'3',280,320)
cmp.calc_nt(pair_10aL10bB,'4',320,430)
cmp.calc_nt(pair_10aL10bB,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.104779	0.623991	0.0418206	0	0	100	0.256863
0.159261	0.219812	0.121455	1	100	200	0.166843
0.13011	0.334182	0.14737	2	200	280	0.203887
0.302609	0.549302	0.0878002	3	280	320	0.313237
0.151841	0.222763	0.142153	4	320	430	0.172253
0.153039	0.372392	0.109921	5	0	430	0.211784


In [10]:
## significant results
#10aL10bB_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10aL10bB,'0',0,100)
cmp.calc(pair_10aL10bB,'1',100,200)
cmp.calc(pair_10aL10bB,'2',200,280)
cmp.calc(pair_10aL10bB,'3',280,320)
cmp.calc(pair_10aL10bB,'4',320,430)
cmp.calc(pair_10aL10bB,'5',0,430)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0703199	0	0	0	0	100	0.02344
0.103744	0	0	1	100	200	0.0345812
0.103593	0	0	2	200	280	0.0345311
0.302609	0	0	3	280	320	0.10087
0.0474133	0	0.068469	4	320	430	0.0386274
0.105296	0	0.0175153	5	0	430	0.0409372


## Figure 10 pair (c)

In [2]:
## calculate pair Gondwana and Laurussia (I-error corrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurussia_RM_GMAP2015AFRcoord","/tmp/Laurussia_RM_GMAP2015AFRcoord")
copy_tree("data/Gondwana_RM","/tmp/Gondwana_RM")
cmp.spa_angpre_len_dif('data/rot_version/Laurussia_RM_GMAP2015AFRcoord.txt',
                       'data/Gondwana_RM.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LauruRMgmap2015AFRcoord_GonRM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	3.9209716311135963	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	2.923070336235969	0	nan	nan	0.20949366544288406	0	104.121754999	336.727180306	1.3865707946920662	1.5960644601349503
2	20.0	3.4535080673406227	0	72.87133315618001	0	0.6560356732675037	0	-48.616204214200025	24.25512894197999	3.4814463776676807	2.825410704400177
3	30.0	3.0085830415898305	0	104.95835750380002	0	0.0344466	0	81.81912663579999	-23.139230868000027	2.9316817046579016	2.8972350920285095
4	40.0	6.708841718277162	0	16.66452436909998	0	2.3236914871705427	0	-22.0037360691	-5.339211700000021	2.0134271947293167	4.3371186818998595
5	50.0	4.687857595481811	0	104.25899320419998	0	4.697997764202508	0	-27.024318764899988	77.2346744393	6.340342837162063	1.6423450729595543
6	60.0	5.795266709596411	1	0.378223292499996	0	9.907339666940917	1	33.4424536082	33.82067690069999	

array([( 0,   0.,  3.92097163, 1,          nan, nan,        nan, nan,             nan,           nan,         nan,         nan),
       ( 1,  10.,  2.92307034, 0,          nan, nan, 0.20949367,  0.,  1.04121755e+02,  336.72718031,  1.38657079,  1.59606446),
       ( 2,  20.,  3.45350807, 0,  72.87133316,  0., 0.65603567,  0., -4.86162042e+01,   24.25512894,  3.48144638,  2.8254107 ),
       ( 3,  30.,  3.00858304, 0, 104.9583575 ,  0., 0.0344466 ,  0.,  8.18191266e+01,  -23.13923087,  2.9316817 ,  2.89723509),
       ( 4,  40.,  6.70884172, 0,  16.66452437,  0., 2.32369149,  0., -2.20037361e+01,   -5.3392117 ,  2.01342719,  4.33711868),
       ( 5,  50.,  4.6878576 , 0, 104.2589932 ,  0., 4.69799776,  0., -2.70243188e+01,   77.23467444,  6.34034284,  1.64234507),
       ( 6,  60.,  5.79526671, 1,   0.37822329,  0., 9.90733967,  1.,  3.34424536e+01,   33.8206769 ,  2.80620434, 12.71354401),
       ( 7,  70.,  3.81561206, 1,  73.35123347,  0., 1.81774504,  0.,  4.20364250e+01,  -31.31480

In [4]:
## reproduce d_notest results for pair (c) in Figure 10(c) using the output above
## from the function cmp.spa_angpre_len_dif
pair_10c=cmp.ppf3('data/0.result_tables/LauruRMgmap2015AFRcoord_GonRM.txt')
#10c_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10c,'0',0,200)
cmp.calc_nt(pair_10c,'1',200,320)
cmp.calc_nt(pair_10c,'2',250,300)
cmp.calc_nt(pair_10c,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.109626	0.378304	0.132781	0	0	200	0.206904
0.143271	0.302843	0.109745	1	200	320	0.185286
0.128525	0.463788	0.119459	2	250	300	0.237258
0.123099	0.35217	0.124142	3	0	320	0.199804


In [5]:
## significant results
#10c_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10c,'0',0,200)
cmp.calc(pair_10c,'1',200,320)
cmp.calc(pair_10c,'2',250,300)
cmp.calc(pair_10c,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.02788	0	0.0183608	0	0	200	0.0154136
0.0923885	0	0	1	200	320	0.0307962
0.0927611	0	0	2	250	300	0.0309204
0.0541373	0	0.0114755	3	0	320	0.0218709


## Figure 10 pair (d)

In [3]:
## calculate pair Gondwana and Laurussia's difference scores in terms of space (coeval poles),
## length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurussia_F6_RM_GMAP2015AFRcoord","/tmp/Laurussia_F6_RM_GMAP2015AFRcoord")
copy_tree("data/Gondwana_F6_RM","/tmp/Gondwana_F6_RM")
cmp.spa_angpre_len_dif('data/rot_version/Laurussia_F6_RM_GMAP2015AFRcoord.txt',
                       'data/Gondwana_F6_RM.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LauruF6RMgmap2015AFRcoord_GonF6RM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	3.080020996588883	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.1789780715118492	0	nan	nan	1.0961851623054866	0	98.7616384223	317.905412915	1.8646867964967404	0.768501634191254
2	20.0	2.6518793211637584	0	101.87501125341998	0	1.7901085075634389	0	-62.44383090957999	39.43118034384	3.2918822931761658	1.501773785612727
3	30.0	1.2994941926563437	0	128.7507706435	0	0.7798868675393249	0	92.2005280465	-36.55024259699999	1.9761147379270099	2.7560016054663348
4	40.0	6.105374347249766	0	31.299042656300003	0	1.8130781516300853	0	-16.08364257330001	15.215400082999992	2.8163868193386588	4.629464970968744
5	50.0	6.788319603666374	1	92.72399001190001	0	4.0541584141548785	0	-29.181761119000015	63.5422288929	6.196125633664609	2.1419672195097306
6	60.0	4.159621900080581	1	7.925520201799998	0	10.653621670992345	1	28.616422325499997	36.5419425272

array([( 0,   0.,  3.080021  , 0,          nan, nan,         nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  1.17897807, 0,          nan, nan,  1.09618516,  0.,   98.76163842,  317.90541291,  1.8646868 ,  0.76850163),
       ( 2,  20.,  2.65187932, 0, 101.87501125,  0.,  1.79010851,  0.,  -62.44383091,   39.43118034,  3.29188229,  1.50177379),
       ( 3,  30.,  1.29949419, 0, 128.75077064,  0.,  0.77988687,  0.,   92.20052805,  -36.5502426 ,  1.97611474,  2.75600161),
       ( 4,  40.,  6.10537435, 0,  31.29904266,  0.,  1.81307815,  0.,  -16.08364257,   15.21540008,  2.81638682,  4.62946497),
       ( 5,  50.,  6.7883196 , 1,  92.72399001,  0.,  4.05415841,  0.,  -29.18176112,   63.54222889,  6.19612563,  2.14196722),
       ( 6,  60.,  4.1596219 , 1,   7.9255202 ,  0., 10.65362167,  1.,   28.61642233,   36.54194253,  2.77843515, 13.43205682),
       ( 7,  70.,  2.36659788, 0,  64.44694291,  0.,  1.91381152,  0.,   40.63004869,  -23.81689421,  3.

In [6]:
## reproduce d_notest results for pair (d) in Figure 10(d) using the output above
## from the function cmp.spa_angpre_len_dif
pair_10d=cmp.ppf3('data/0.result_tables/LauruF6RMgmap2015AFRcoord_GonF6RM.txt')
#10d_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10d,'0',0,200)
cmp.calc_nt(pair_10d,'1',200,320)
cmp.calc_nt(pair_10d,'2',250,300)
cmp.calc_nt(pair_10d,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.101458	0.414845	0.130731	0	0	200	0.215678
0.0954765	0.383661	0.128712	1	200	320	0.202617
0.100761	0.480246	0.0910353	2	250	300	0.224014
0.100264	0.398698	0.129974	3	0	320	0.209645


In [7]:
## significant results
#10d_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10d,'0',0,200)
cmp.calc(pair_10d,'1',200,320)
cmp.calc(pair_10d,'2',250,300)
cmp.calc(pair_10d,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0308188	0	0.0197438	0	0	200	0.0168542
0.0311582	0	0	1	200	320	0.0103861
0.0324926	0	0	2	250	300	0.0108309
0.0318864	0	0.0123399	3	0	320	0.0147421


## Figure 10 pair (c)'s Gondwana (Uncorrected) vs pair (d)'s Laurussia (Corrected)

In [5]:
## calculate pair Gondwana (Uncorrected) and Laurussia (Corrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurussia_F6_RM_GMAP2015AFRcoord","/tmp/Laurussia_F6_RM_GMAP2015AFRcoord")
copy_tree("data/Gondwana_RM","/tmp/Gondwana_RM")
cmp.spa_angpre_len_dif('data/rot_version/Laurussia_F6_RM_GMAP2015AFRcoord.txt',
                       'data/Gondwana_RM.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LauruF6RMgmap2015AFRcoord_GonRM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	3.080020996588883	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.549773250676106	0	nan	nan	0.2686223363617901	0	98.7616384223	336.727180306	1.8646867964967404	1.5960644601349503
2	20.0	2.494365061057607	0	86.69895985155998	0	0.4664715887759887	0	-62.44383090957999	24.25512894197999	3.2918822931761658	2.825410704400177
3	30.0	3.0085830415898305	0	115.33975891450002	0	0.9211203541014996	0	92.2005280465	-23.139230868000027	1.9761147379270099	2.8972350920285095
4	40.0	7.114946556426819	0	10.74443087329999	0	1.5207318625612007	0	-16.08364257330001	-5.339211700000021	2.8163868193386588	4.3371186818998595
5	50.0	6.289108295930069	0	106.41643555830001	0	4.5537805607050545	0	-29.181761119000015	77.2346744393	6.196125633664609	1.6423450729595543
6	60.0	4.159621900080581	1	5.204254575199997	0	9.93510886369306	1	28.616422325499997	33.82067

array([( 0,   0.,  3.080021  , 0,          nan, nan,         nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  1.54977325, 0,          nan, nan,  0.26862234,  0.,   98.76163842,  336.72718031,  1.8646868 ,  1.59606446),
       ( 2,  20.,  2.49436506, 0,  86.69895985,  0.,  0.46647159,  0.,  -62.44383091,   24.25512894,  3.29188229,  2.8254107 ),
       ( 3,  30.,  3.00858304, 0, 115.33975891,  0.,  0.92112035,  0.,   92.20052805,  -23.13923087,  1.97611474,  2.89723509),
       ( 4,  40.,  7.11494656, 0,  10.74443087,  0.,  1.52073186,  0.,  -16.08364257,   -5.3392117 ,  2.81638682,  4.33711868),
       ( 5,  50.,  6.2891083 , 0, 106.41643556,  0.,  4.55378056,  0.,  -29.18176112,   77.23467444,  6.19612563,  1.64234507),
       ( 6,  60.,  4.1596219 , 1,   5.20425458,  0.,  9.93510886,  1.,   28.61642233,   33.8206769 ,  2.77843515, 12.71354401),
       ( 7,  70.,  2.36659788, 0,  71.94485712,  0.,  1.91381152,  0.,   40.63004869,  -31.31480843,  3.

In [13]:
pair_10cG10dL=cmp.ppf3('data/0.result_tables/LauruF6RMgmap2015AFRcoord_GonRM.txt')
#10cG10dL_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10cG10dL,'0',0,200)
cmp.calc_nt(pair_10cG10dL,'1',200,320)
cmp.calc_nt(pair_10cG10dL,'2',250,300)
cmp.calc_nt(pair_10cG10dL,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.105796	0.408388	0.127912	0	0	200	0.214032
0.117145	0.288145	0.119261	1	200	320	0.17485
0.102873	0.451105	0.12483	2	250	300	0.226269
0.110772	0.362577	0.124668	3	0	320	0.199339


In [14]:
## significant results
#10cG10dL_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10cG10dL,'0',0,200)
cmp.calc(pair_10cG10dL,'1',200,320)
cmp.calc(pair_10cG10dL,'2',250,300)
cmp.calc(pair_10cG10dL,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0243537	0	0.0184122	0	0	200	0.0142553
0.0394146	0	0	1	200	320	0.0131382
0.0295949	0	0	2	250	300	0.00986498
0.0310248	0	0.0115077	3	0	320	0.0141775


## Figure 10 pair (c)'s Laurussia (Uncorrected) vs pair (d)'s Gondwana (Corrected)

In [6]:
## calculate pair Gondwana (Corrected) and Laurussia (Uncorrected)'s difference scores in terms of
## space (coeval poles), length (coeval segments) and angle (coeval orientation change)
copy_tree("data/rot_version/Laurussia_RM_GMAP2015AFRcoord","/tmp/Laurussia_RM_GMAP2015AFRcoord")
copy_tree("data/Gondwana_F6_RM","/tmp/Gondwana_F6_RM")
cmp.spa_angpre_len_dif('data/rot_version/Laurussia_RM_GMAP2015AFRcoord.txt',
                       'data/Gondwana_F6_RM.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/LauruRMgmap2015AFRcoord_GonF6RM.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	3.9209716311135963	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	2.5185444395338887	0	nan	nan	0.6180691605008123	0	104.121754999	317.905412915	1.3865707946920662	0.768501634191254
2	20.0	3.239031150781701	0	88.04738455804002	0	1.9796725920549538	0	-48.616204214200025	39.43118034384	3.4814463776676807	1.501773785612727
3	30.0	1.2994941926563437	0	118.36936923279998	0	0.1756800991915668	0	81.81912663579999	-36.55024259699999	2.9316817046579016	2.7560016054663348
4	40.0	5.452613672290489	0	37.21913615209999	0	2.6160377762394273	0	-22.0037360691	15.215400082999992	2.0134271947293167	4.629464970968744
5	50.0	5.287172335509594	0	90.56654765779999	0	4.198375617652332	0	-27.024318764899988	63.5422288929	6.340342837162063	2.1419672195097306
6	60.0	5.795266709596411	1	3.099488919099997	0	10.625852474240203	1	33.4424536082	36.5419425272999

array([( 0,   0.,  3.92097163, 1,          nan, nan,         nan, nan,             nan,           nan,         nan,         nan),
       ( 1,  10.,  2.51854444, 0,          nan, nan,  0.61806916,  0.,  1.04121755e+02,  317.90541291,  1.38657079,  0.76850163),
       ( 2,  20.,  3.23903115, 0,  88.04738456,  0.,  1.97967259,  0., -4.86162042e+01,   39.43118034,  3.48144638,  1.50177379),
       ( 3,  30.,  1.29949419, 0, 118.36936923,  0.,  0.1756801 ,  0.,  8.18191266e+01,  -36.5502426 ,  2.9316817 ,  2.75600161),
       ( 4,  40.,  5.45261367, 0,  37.21913615,  0.,  2.61603778,  0., -2.20037361e+01,   15.21540008,  2.01342719,  4.62946497),
       ( 5,  50.,  5.28717234, 0,  90.56654766,  0.,  4.19837562,  0., -2.70243188e+01,   63.54222889,  6.34034284,  2.14196722),
       ( 6,  60.,  5.79526671, 1,   3.09948892,  0., 10.62585247,  1.,  3.34424536e+01,   36.54194253,  2.80620434, 13.43205682),
       ( 7,  70.,  3.81561206, 0,  65.85331926,  0.,  1.81774504,  0.,  4.20364250e+01,  -

In [15]:
pair_10cL10dG=cmp.ppf3('data/0.result_tables/LauruRMgmap2015AFRcoord_GonF6RM.txt')
#10cL10dG_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pair_10cL10dG,'0',0,200)
cmp.calc_nt(pair_10cL10dG,'1',200,320)
cmp.calc_nt(pair_10cL10dG,'2',250,300)
cmp.calc_nt(pair_10cL10dG,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.104801	0.384761	0.135346	0	0	200	0.208303
0.091046	0.371167	0.113494	1	200	320	0.191902
0.083295	0.477013	0.0856651	2	250	300	0.215324
0.10022	0.378642	0.127152	3	0	320	0.202005


In [16]:
## significant results
#10cL10dG_test.txt
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pair_10cL10dG,'0',0,200)
cmp.calc(pair_10cL10dG,'1',200,320)
cmp.calc(pair_10cL10dG,'2',250,300)
cmp.calc(pair_10cL10dG,'3',0,320)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0281006	0	0.0196924	0	0	200	0.015931
0.0152195	0	0	1	200	320	0.00507315
0.0329755	0	0	2	250	300	0.0109918
0.0238777	0	0.0123077	3	0	320	0.0120618


## pair v-i (uninterpolated; bin 8 Myr, step 4 Myr)

In [ ]:
## calculate pair v-i (uninterpolated; bin 8 step 4)'s difference scores in terms of
## space (coeval poles), angle and length (coeval segments)
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_stp4_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_stp4.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in data/0.result_tables/stp4.txt

In [22]:
## reproduce d_notest results for pair v-i (uninterpolated; bin 8 step 4) in Figure
## 11a using the output above from the function cmp.spa_angpre_len_dif
pair_v_i_stp4=cmp.ppf3('data/0.result_tables/stp4.txt')
#v_i stp4_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pair_v_i_stp4,0,0,100)
cmp.calc_nt(pair_v_i_stp4,1,100,200)
cmp.calc_nt(pair_v_i_stp4,2,200,280)
cmp.calc_nt(pair_v_i_stp4,3,280,320)
cmp.calc_nt(pair_v_i_stp4,4,320,432)
cmp.calc_nt(pair_v_i_stp4,5,0,432)

ds	da	dl	no	t0	t1
0.14101349557151108	0.5247688276161536	0.20612139692150436	0	0	100
0.20297305051719985	0.3634743520500222	0.24661701866046506	1	100	200
0.137961759115263	0.4142267703688271	0.3586361600698007	2	200	280
0.2798104052617373	0.5234872479065209	0.26449830440346284	3	280	320
0.16900667438700645	0.5714621622488194	0.08206255168904339	4	320	432
0.17761638644017358	0.4572953237546528	0.21706487492383217	5	0	432


In [23]:
## reproduce d_tested results for pair v-i (uninterpolated; bin 8 step 4) in Figure
## 11a using the same output above from the function cmp.spa_angpre_len_dif
#v_i stp4_test.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pair_v_i_stp4,0,0,100)
cmp.calc(pair_v_i_stp4,1,100,200)
cmp.calc(pair_v_i_stp4,2,200,280)
cmp.calc(pair_v_i_stp4,3,280,320)
cmp.calc(pair_v_i_stp4,4,320,432)
cmp.calc(pair_v_i_stp4,5,0,432)

ds	da	dl	no	t0	t1
0.07089586421508128	0.0	0.0	0	0	100
0.11845957944546535	0.0	0.036389452384814995	1	100	200
0.04195924779303895	0.0	0.032186433697426915	2	200	280
0.2622782694793313	0.0	0.07203052825801613	3	280	320
0.13862466446386373	0.0	0.0	4	320	432
0.11093308872908357	0.0	0.02582081487119092	5	0	432


## pair v-i (uninterpolated; bin 12 Myr, step 6 Myr)

In [ ]:
## calculate pair v-i (uninterpolated; bin 12 step 6)'s difference scores in terms of
## space (coeval poles), angle and length (coeval segments)
copy_tree("data/Baltica_StableEurope_F6_RM_stp6","/tmp/Baltica_StableEurope_F6_RM_stp6")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_stp6_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_stp6.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in data/0.result_tables/stp6.txt

In [24]:
## reproduce d_notest results for pair v-i (uninterpolated; bin 12 step 6) in Figure
## 11b using the output above from the function cmp.spa_angpre_len_dif
pair_v_i_stp6=cmp.ppf3('data/0.result_tables/stp6.txt')
#v_i stp6_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pair_v_i_stp6,0,0,102)
cmp.calc_nt(pair_v_i_stp6,1,102,198)
cmp.calc_nt(pair_v_i_stp6,2,198,282)
cmp.calc_nt(pair_v_i_stp6,3,282,324)
cmp.calc_nt(pair_v_i_stp6,4,324,432)
cmp.calc_nt(pair_v_i_stp6,5,0,432)

ds	da	dl	no	t0	t1
0.13479526029554387	0.4867079097798532	0.16696593446115385	0	0	102
0.17516323730799446	0.28708755113419193	0.1784577464625825	1	102	198
0.11446606473210975	0.38349057501707273	0.1546803621316973	2	198	282
0.2863871758464745	0.45186958304647773	0.16868394721492916	3	282	324
0.13790886848679304	0.14300401581240743	0.13360093644690493	4	324	432
0.15905167428991887	0.3773495158281903	0.16373516116353126	5	0	432


In [25]:
## reproduce d_tested results for pair v-i (uninterpolated; bin 12 step 6) in Figure
## 11b using the same output above from the function cmp.spa_angpre_len_dif
#v_i stp6_test.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pair_v_i_stp6,0,0,102)
cmp.calc(pair_v_i_stp6,1,102,198)
cmp.calc(pair_v_i_stp6,2,198,282)
cmp.calc(pair_v_i_stp6,3,282,324)
cmp.calc(pair_v_i_stp6,4,324,432)
cmp.calc(pair_v_i_stp6,5,0,432)

ds	da	dl	no	t0	t1
0.0796366211957521	0.0	0.0	0	0	102
0.09507983827926063	0.0	0.034369626045534236	1	102	198
0.026787580509494133	0.0	0.0	2	198	282
0.2863871758464745	0.0	0.0	3	282	324
0.043346964380049764	0.0	0.044544736735248724	4	324	432
0.09020193986950878	0.0	0.027132175597644026	5	0	432


## pair v-i (uninterpolated; bin 16 Myr, step 8 Myr)

In [ ]:
## calculate pair v-i (uninterpolated; bin 16 step 8)'s difference scores in terms of
## space (coeval poles), angle and length (coeval segments)
copy_tree("data/Baltica_StableEurope_F6_RM_stp8","/tmp/Baltica_StableEurope_F6_RM_stp8")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_stp8_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_stp8.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in data/0.result_tables/stp8.txt

In [26]:
## reproduce d_notest results for pair v-i (uninterpolated; bin 16 step 8) in Figure
## 11c using the output above from the function cmp.spa_angpre_len_dif
pair_v_i_stp8=cmp.ppf3('data/0.result_tables/stp8.txt')
#v_i stp8_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pair_v_i_stp8,0,0,104)
cmp.calc_nt(pair_v_i_stp8,1,104,200)
cmp.calc_nt(pair_v_i_stp8,2,200,280)
cmp.calc_nt(pair_v_i_stp8,3,280,320)
cmp.calc_nt(pair_v_i_stp8,4,320,432)
cmp.calc_nt(pair_v_i_stp8,5,0,432)

ds	da	dl	no	t0	t1
0.11215826253805945	0.5138513685563888	0.10712709007575096	0	0	104
0.16511983276071432	0.35231170481395074	0.21916664555662618	1	104	200
0.10967789212354384	0.3967831561728149	0.13890111882128364	2	200	280
0.3081426483535621	0.42848785261708316	0.1802621667260313	3	280	320
0.16813929604791303	0.375796316450635	0.13350135644622085	4	320	432
0.15359673256539183	0.40000166800647285	0.15151838758804373	5	0	432


In [27]:
## reproduce d_tested results for pair v-i (uninterpolated; bin 16 step 8) in Figure
## 11c using the same output above from the function cmp.spa_angpre_len_dif
#v_i stp8_test.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pair_v_i_stp8,0,0,104)
cmp.calc(pair_v_i_stp8,1,104,200)
cmp.calc(pair_v_i_stp8,2,200,280)
cmp.calc(pair_v_i_stp8,3,280,320)
cmp.calc(pair_v_i_stp8,4,320,432)
cmp.calc(pair_v_i_stp8,5,0,432)

ds	da	dl	no	t0	t1
0.047911641984121636	0.0	0.0	0	0	104
0.11956535291688482	0.0	0.04188406190880708	1	104	200
0.021298455611132472	0.0	0.0	2	200	280
0.3081426483535621	0.0	0.0	3	280	320
0.0947402639756929	0.0	0.053967404491233904	4	320	432
0.09117627892481336	0.0	0.023299118625610366	5	0	432


## pair v-i (uninterpolated; bin 20 Myr, step 10 Myr)

In [ ]:
## calculate pair v-i (uninterpolated)'s difference scores in terms of
## space (coeval poles), angle and length (coeval segments)
copy_tree("data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord","/tmp/Laurentia_F6_RM_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_F6_RM","/tmp/Baltica_StableEurope_F6_RM")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in
# data/0.result_tables/v_i_LaureF6RMgmap2015BALcoord_BalF6RM.txt

In [28]:
## reproduce d_notest results for pair v-i (uninterpolated) in Figure
## 11d using the output above from the function cmp.spa_angpre_len_dif
pair_v_i_stp10=cmp.ppf3('data/0.result_tables/v_i_LaureF6RMgmap2015BALcoord_BalF6RM.txt')
#v_i stp10_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pair_v_i_stp10,0,0,100)
cmp.calc_nt(pair_v_i_stp10,1,100,200)
cmp.calc_nt(pair_v_i_stp10,2,200,280)
cmp.calc_nt(pair_v_i_stp10,3,280,320)
cmp.calc_nt(pair_v_i_stp10,4,320,430)
cmp.calc_nt(pair_v_i_stp10,5,0,430)

ds	da	dl	no	t0	t1
0.10799163338819137	0.5831435834757162	0.06039215698680937	0	0	100
0.14088555971381247	0.42435339088256957	0.1571594779742826	1	100	200
0.10612454213309036	0.365272484455754	0.14523851880936933	2	200	280
0.28996316838673214	0.5651518843799963	0.10986373583020166	3	280	320
0.13864867060086583	0.15780966540037025	0.15275806228831662	4	320	430
0.13975708066709464	0.3957881760300837	0.12691204950181778	5	0	430


In [29]:
## reproduce d_tested results for pair v-i (uninterpolated) in Figure
## 11d using the same output above from the function cmp.spa_angpre_len_dif
#v_i stp10_test.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pair_v_i_stp10,0,0,100)
cmp.calc(pair_v_i_stp10,1,100,200)
cmp.calc(pair_v_i_stp10,2,200,280)
cmp.calc(pair_v_i_stp10,3,280,320)
cmp.calc(pair_v_i_stp10,4,320,430)
cmp.calc(pair_v_i_stp10,5,0,430)

ds	da	dl	no	t0	t1
0.04419549615754297	0.0	0.0	0	0	100
0.06604746085768656	0.0	0.03237979956760407	1	100	200
0.02658221857196827	0.0	0.0	2	200	280
0.28996316838673214	0.0	0.0	3	280	320
0.05652227149518458	0.0	0.06657413517909282	4	320	430
0.06913839372806609	0.0	0.024560778666187482	5	0	430


## pair v-i (uninterpolated; bin 24 Myr, step 12 Myr)

In [ ]:
## calculate pair v-i (uninterpolated; bin 24 step 12)'s difference scores in terms of
## space (coeval poles), angle and length (coeval segments)
copy_tree("data/rot_version/Laurentia_F6_RM_stp12_GMAP2015BALcoord","/tmp/Laurentia_F6_RM_stp12_GMAP2015BALcoord")
copy_tree("data/Baltica_StableEurope_F6_RM_stp12","/tmp/Baltica_StableEurope_F6_RM_stp12")
cmp.spa_angpre_len_dif('data/rot_version/Laurentia_F6_RM_stp12_GMAP2015BALcoord.txt',
                       'data/Baltica_StableEurope_F6_RM_stp12.txt',
                       pnh1=0,pnh2=1)
# the results should be the same as stored in data/0.result_tables/stp12.txt

In [30]:
## reproduce d_notest results for pair v-i (uninterpolated; bin 24 step 12) in Figure
## 11f using the output above from the function cmp.spa_angpre_len_dif
pair_v_i_stp12=cmp.ppf3('data/0.result_tables/stp12.txt')
#v_i stp12_ntest.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pair_v_i_stp12,0,0,96)
cmp.calc_nt(pair_v_i_stp12,1,96,204)
cmp.calc_nt(pair_v_i_stp12,2,204,276)
cmp.calc_nt(pair_v_i_stp12,3,276,324)
cmp.calc_nt(pair_v_i_stp12,4,324,432)
cmp.calc_nt(pair_v_i_stp12,5,0,432)

ds	da	dl	no	t0	t1
0.1028850067785222	0.46678587624615875	0.046231169856320024	0	0	96
0.1292373168031566	0.3091686989006528	0.17928963066819184	1	96	204
0.10165590888284332	0.30191875889277764	0.07834646332609962	2	204	276
0.23071352626479374	0.6507660583145556	0.14480401144417307	3	276	324
0.12330830121123898	0.16951691690000004	0.06556327001473632	4	324	432
0.13281800898076074	0.3650317824311219	0.1006338969647279	5	0	432


In [31]:
## reproduce d_tested results for pair v-i (uninterpolated; bin 24 step 12) in Figure
## 11f using the same output above from the function cmp.spa_angpre_len_dif
#v_i stp12_test.txt
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pair_v_i_stp12,0,0,96)
cmp.calc(pair_v_i_stp12,1,96,204)
cmp.calc(pair_v_i_stp12,2,204,276)
cmp.calc(pair_v_i_stp12,3,276,324)
cmp.calc(pair_v_i_stp12,4,324,432)
cmp.calc(pair_v_i_stp12,5,0,432)

ds	da	dl	no	t0	t1
0.02810811353344097	0.0	0.0	0	0	96
0.053439917581842876	0.0	0.05101169608468492	1	96	204
0.06325661153384379	0.0	0.0	2	204	276
0.21382472114043094	0.0	0.0	3	276	324
0.0	0.0	0.0	4	324	432
0.06372734435578983	0.0	0.01275292402117123	5	0	432


# Random Spacial Uncertainty Cases Based on pair C

The 4th and 5th columns in data/Laurentia_RM.txt and data/Laurentia_RM_NoIntp.txt are dm and dp or A95. There are 49 dm, dp and A95 values (interpolated mean poles got zero uncertainty).

    2.1773837818506814
    2.2431473270377436
    3.71750970800317
    3.1023530620806334
    2.8113017709316428
    4.0134175890381663
    3.786531931743133
    4.5900107578669189
    6.9666509571459523
    6.2557646831508871
    13.141908846033042
    4.7347372014471407
    3.2057231363612635
    4.0213091346077938
    18.271260010761665
    12.517051610516857
    9.1878895238294351
    10.590469280126225
    6.8263580723427415
    3.4874695224517662
    2.8244769407192094
    2.2721366843710524
    2.5041569487865676
    3.3001616168940293
    4.3553494144442579
    12.536533258858929
    3.4473456784553647
    4.1990565047872881
    4.8146324156324543
    3.8476198382425539
    4.3540409528317152
    6.9558423063127517
    9.4254454138432369
    8.8446407802130125
    10.948879612191245
    16.
    11.39999962
    5.9
    19.183628069179896
    19.258717698236371
    17.534201407505424
    7.3
    11.123232589
    14.720652226
    7.847140921
    6.22406599
    12.912932754
    6.9
    3.8

Based on these values, the distribution is identified through "fitter" package.

In [1]:
from fitter import Fitter

f = Fitter([2.17738,2.24315,3.71751,3.10235,2.8113,4.01342,3.78653,4.59001,6.96665,6.25576,13.1419,4.73474,3.20572,4.02131,18.2713,12.5171,9.18789,10.5905,6.82636,3.48747,2.82448,2.27214,2.50416,3.30016,4.35535,12.5365,3.44735,4.19906,4.81463,3.84762,4.35404,6.95584,9.42545,8.84464,10.9489,16,11.4,5.9,19.1836,19.2587,17.5342,7.3,11.1232,14.7207,7.84714,6.22407,12.9129,6.9,3.8])

f.fit()

f.get_best()

/home/g/conda/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py:2316: RuntimeWarning: invalid value encountered in double_scalars
  Lhat = muhat - Shat*mu
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:390: RuntimeWarning: divide by zero encountered in true_divide
  return 1.0/np.pi/np.sqrt(x*(1-x))


Fitted alpha distribution with error=0.8054883204060229)
Fitted anglit distribution with error=0.9783152028002629)
Fitted arcsine distribution with error=0.9800587959867477)
Fitted argus distribution with error=1.154706248227962)
Fitted beta distribution with error=0.7667207751926125)
Fitted betaprime distribution with error=0.7909193629623349)
Fitted bradford distribution with error=0.7650505445992828)
Fitted burr distribution with error=0.7876811565832144)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:515: RuntimeWarning: invalid value encountered in sqrt
  sk = 2*(b-a)*np.sqrt(a + b + 1) / (a + b + 2) / np.sqrt(a*b)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:667: RuntimeWarning: divide by zero encountered in true_divide
  a/(b-1.0),
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:671: RuntimeWarning: divide by zero encountered in true_divide
  a*(a+1.0)/((b-2.0)*(b-1.0)),
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py:1035: RuntimeWarning: invalid value encountered in subtract
  mu2 = mu2p - mu * mu


Fitted burr12 distribution with error=0.7892562247711464)
Fitted cauchy distribution with error=0.8884916404818229)
Fitted chi distribution with error=0.7499172215925255)
Fitted chi2 distribution with error=0.8105384856608353)
Fitted cosine distribution with error=0.9651196943618614)
Fitted crystalball distribution with error=0.9480924680188824)
Fitted dgamma distribution with error=0.8539592640471295)
Fitted dweibull distribution with error=0.9109835755431281)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2651: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value 4.2.
  RuntimeWarning)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2651: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value 4.133333333333333.
  RuntimeWarning)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2651: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value 4.222222222222221.
  RuntimeWarning)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2651: RuntimeWarning: The shape parameter of the erlang distribution has been given a non-integer value 4.333333333333332.
  RuntimeWarning)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2651: RuntimeWarning: The shape parameter of the erlang distribution h

Fitted erlang distribution with error=0.9004743519574526)
Fitted expon distribution with error=0.7580789352039634)
Fitted exponnorm distribution with error=0.7580685201697166)
Fitted exponpow distribution with error=0.7539538664100945)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:1469: RuntimeWarning: invalid value encountered in add
  negxc + sc.xlogy(c - 1.0, x))


Fitted exponweib distribution with error=0.9117040583072723)
Fitted f distribution with error=0.7543848882177319)
Fitted fatiguelife distribution with error=0.7753376844194321)
Fitted fisk distribution with error=0.778467129910108)
Fitted foldcauchy distribution with error=0.757924077936547)
Fitted foldnorm distribution with error=0.8130914155969623)
Fitted frechet_l distribution with error=2.1466416913764683)
Fitted frechet_r distribution with error=0.78802929482801)
Fitted gamma distribution with error=0.7521724094362768)
Fitted gausshyper distribution with error=0.8276198251121927)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:3175: RuntimeWarning: divide by zero encountered in power
  return 1.0/Cinv * x**(a-1.0) * (1.0-x)**(b-1.0) / (1.0+z*x)**c


Fitted genexpon distribution with error=0.765454522888268)
Fitted genextreme distribution with error=0.7915351578736389)
Fitted gengamma distribution with error=0.7721391036306233)
Fitted genhalflogistic distribution with error=0.7953108091952219)
Fitted genlogistic distribution with error=0.8798316598476352)
Fitted gennorm distribution with error=0.9562577328650591)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:2248: RuntimeWarning: divide by zero encountered in true_divide
  val = val + cnk * (-1) ** ki / (1.0 - c * ki)


Fitted genpareto distribution with error=0.7739649136897396)
Fitted gilbrat distribution with error=0.7768966566353368)
Fitted gompertz distribution with error=0.7676882870606342)
Fitted gumbel_l distribution with error=1.0070829443472151)
Fitted gumbel_r distribution with error=0.8804797274716167)
Fitted halfcauchy distribution with error=0.755981795649219)
Fitted halfgennorm distribution with error=0.7967603341789506)
Fitted halflogistic distribution with error=0.7948349374983947)
Fitted halfnorm distribution with error=0.8161096776082977)
Fitted hypsecant distribution with error=0.9268536774069684)
Fitted invgamma distribution with error=0.7911227272963223)
Fitted invgauss distribution with error=0.7797764131634809)
Fitted invweibull distribution with error=0.7915393275543918)
Fitted johnsonsb distribution with error=0.7687267560941016)
Fitted johnsonsu distribution with error=0.7797043696958079)
Fitted kappa3 distribution with error=0.7772424183710336)
Fitted kappa4 distribution wi

/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:79: RuntimeWarning: floating point number truncated to an integer
  return -scu._smirnovp(n, x)


Fitted levy_stable distribution with error=0.9127796719776363)
Fitted loggamma distribution with error=0.9476739975014272)
Fitted logistic distribution with error=0.9353142982899302)
Fitted loglaplace distribution with error=0.8983609785224554)
Fitted lognorm distribution with error=0.7797366194914643)
Fitted lomax distribution with error=0.7568135439575983)
Fitted maxwell distribution with error=0.9208622984596819)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:4299: RuntimeWarning: divide by zero encountered in true_divide
  return c**2 / (c**2 - n**2)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py:1618: IntegrationWarning: Extremely bad integrand behavior occurs at some points of the
  integration interval.
  return integrate.quad(self._mom_integ1, 0, 1, args=(m,)+args)[0]
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py:1618: IntegrationWarning: The integral is probably divergent, or slowly convergent.
  return integrate.quad(self._mom_integ1, 0, 1, args=(m,)+args)[0]
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:4578: RuntimeWarning: overflow encountered in power
  return k*x**(k-1.0) / (1.0+x**s)**(1.0+k*1.0/s)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_distn_infrastructure.py:1625: RuntimeWarning: divide by zero encountered in log
  return log(self._pdf(x, *ar

Fitted mielke distribution with error=0.7923637013029168)
Fitted moyal distribution with error=0.8512331624907457)
Fitted nakagami distribution with error=0.7596787312831463)
Fitted ncf distribution with error=0.8041256027470239)
Fitted nct distribution with error=0.789941301357097)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5286: RuntimeWarning: overflow encountered in multiply
  trm1 = np.sqrt(2)*nc*x*sc.hyp1f1(n/2+1, 1.5, valF)


Fitted ncx2 distribution with error=0.8652713697069289)
Fitted norm distribution with error=0.9480924657989177)
Fitted norminvgauss distribution with error=0.7795141168056093)
Fitted pareto distribution with error=0.8761642086398547)
Fitted pearson3 distribution with error=0.7523590631912731)
Fitted powerlaw distribution with error=0.7817321402516422)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5689: RuntimeWarning: divide by zero encountered in power
  pow(_norm_cdf(-np.log(x)/s), c*1.0-1.0))
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5689: RuntimeWarning: invalid value encountered in multiply
  pow(_norm_cdf(-np.log(x)/s), c*1.0-1.0))
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5689: RuntimeWarning: overflow encountered in power
  pow(_norm_cdf(-np.log(x)/s), c*1.0-1.0))
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5726: RuntimeWarning: divide by zero encountered in power
  return c*_norm_pdf(x) * (_norm_cdf(-x)**(c-1.0))
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5726: RuntimeWarning: invalid value encountered in multiply
  return c*_norm_pdf(x) * (_norm_cdf(-x)**(c-1.0))
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:5773: RuntimeWarning: div

Fitted powerlognorm distribution with error=0.7832275547754222)
Fitted powernorm distribution with error=nan)
Fitted rayleigh distribution with error=0.9101218153493682)
Fitted rdist distribution with error=1.0334904035626156)
Fitted recipinvgauss distribution with error=0.7725024579665051)
Fitted reciprocal distribution with error=1.298991695636648)
Fitted rice distribution with error=0.9101233470801018)
SKIPPED rv_continuous distribution (taking more than 30 seconds)
SKIPPED rv_histogram distribution (taking more than 30 seconds)
Fitted semicircular distribution with error=0.9865793958408592)
Fitted skewnorm distribution with error=0.8161156217923726)
Fitted t distribution with error=0.9480949837640171)
Fitted trapz distribution with error=1.1973832109310898)
Fitted triang distribution with error=0.8246262354390209)
Fitted truncexpon distribution with error=0.8689560435836432)
Fitted truncnorm distribution with error=1.298991695636648)


/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:6348: RuntimeWarning: divide by zero encountered in log
  self._logdelta = np.log(self._delta)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:6348: RuntimeWarning: invalid value encountered in log
  self._logdelta = np.log(self._delta)
/home/g/conda/lib/python3.7/site-packages/scipy/stats/_continuous_distns.py:6408: RuntimeWarning: divide by zero encountered in power
  Px = Fx**(lam-1.0) + (np.asarray(1-Fx))**(lam-1.0)


Fitted tukeylambda distribution with error=0.9366334353222865)
Fitted uniform distribution with error=0.9562577328648245)
Fitted vonmises distribution with error=1.3833584804376924e+61)
Fitted vonmises_line distribution with error=0.9253542704201618)
Fitted wald distribution with error=0.7781082263191528)
Fitted weibull_max distribution with error=2.1466416913764683)
Fitted weibull_min distribution with error=0.78802929482801)
Fitted wrapcauchy distribution with error=nan)


/home/g/conda/lib/python3.7/site-packages/scipy/optimize/optimize.py:570: RuntimeWarning: invalid value encountered in subtract
  numpy.max(numpy.abs(fsim[0] - fsim[1:])) <= fatol):


{'chi': (0.6994672964288513, 2.1773799999999994, 8.624383899041122)}

So the distribution of the 49 uncertainty values is close to a Chi distribution (degrees of freedom [df]: 0.6994672964288513; shift [loc]: 2.17738; scale: 8.624383899041122). Then we use the same "df" and "loc" parameters from this Chi distribution, and a different "scale" parameter, e.g. here 2, to regenerate 49 random variates (2 mean poles own actually oval spacial uncertainties).

In [4]:
from scipy import stats

#rvs(df, loc=0, scale=1, size=1, random_state=None)	Random variates.
#To shift and/or scale the distribution use the loc and scale parameters.
stats.chi.rvs(0.6994672964288513, loc=2.1773799999999994, scale=2, size=49)
#every time you run this line, you get different sets of values. I got:

array([3.34768645, 5.74106438, 2.98062202, 2.81983593, 4.5748303 ,
       4.45594926, 2.3146175 , 2.78750108, 2.29648684, 3.68447742,
       2.26313146, 3.20705208, 2.18819118, 2.45410824, 4.29518793,
       2.50129803, 3.70427124, 2.30540412, 2.25618714, 2.68798882,
       2.87903861, 3.35817813, 3.28486568, 3.55397227, 3.77051584,
       3.44572636, 2.41503632, 4.92909196, 5.19927303, 2.60833878,
       3.93066969, 3.12165022, 2.93741435, 2.30992913, 2.58564085,
       4.33245737, 2.60947676, 2.26452186, 2.17740061, 2.48414874,
       3.60984599, 6.08072785, 2.64560573, 2.64028049, 5.05128373,
       2.70394219, 3.02698954, 2.73149277, 2.17742105])

Their azimuths for dm (semi-major axis) are randomly derived from -180 to 180.

In [5]:
import random

for i in range(47):
    print(random.uniform(-180,180))

-84.72106956847605
-101.67129380485784
-9.908942198398535
-155.61561918364177
-170.36597906590882
30.63018866620851
-78.84581679275597
134.0124349726517
23.266315427792307
-158.95240647211077
161.97308974176053
84.34176986714999
176.42344557623835
20.744878806802888
-166.806699218913
139.64225953183643
72.38494527694868
22.57207231148334
87.39496610789644
43.16046830066307
37.81815181449781
-11.8669793965353
-144.9509258977978
151.51046216163576
179.9324729169341
78.6313530511473
-95.7881310357444
-104.39448099531899
-158.68250793466308
99.94164126774814
-92.94616105506496
-59.701741651557
-47.95273336831514
144.68462876734742
4.924119906921305
31.349376766926127
-153.88034550770487
11.786332659454843
-90.21439682714195
4.978574247163493
37.814221660250325
-110.02560265399627
144.16335529334316
-165.35684482740345
-42.424739108462944
36.880291808951256
17.80332730716131
-168.35450723995373
112.36410106795745
82.27700471064878
169.81806129197668
-88.07628120065495
121.9648045040139
10.1

Then do not forget to change the "Number of Paleopoles" column to 0 or 1, in order to use dm/dp or A95 to generate random points for testing (see the function "common_dir_elliptical").

## pair H

In [5]:
copy_tree("data/Laurentia_RM", "/tmp/Laurentia_RM")

['/tmp/Laurentia_RM/210.0.txt', '/tmp/Laurentia_RM/220.0.txt']

In [7]:
## calculate pair H's difference scores in terms of space (coeval poles), angle and
## length (coeval segments)
cmp.spa_angpre_len_dif('data/Laurentia_RM.txt',
                       'data/rot_version/Laurentia_RM_po_15deg_re.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairH_LaurentiaRM_LaurentiaRMpo15degRe.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	1.0396087803797036	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.0488574183293418	0	nan	nan	0.000005	0	273.743775125	157.610292419	0.5238885648671224	0.523883588268831
2	20.0	1.939251148514321	0	84.64397175799996	0	0.0003675	0	139.00062497750002	-136.3554032645	4.4390170905897985	4.438649608107905
3	30.0	2.186441670519566	0	91.63317308890001	0	0.0001424	0	44.63274003309999	136.265913122	2.1924434711123775	2.192301096824105
4	40.0	2.4982177538741293	0	78.66120356299999	0	0.0000185	0	-76.27212559900002	-154.933329162	1.2697772061265127	1.2697587232430259
5	50.0	3.9097779362449967	0	4.176170072699989	0	0.0016097	0	48.33596726459999	44.1597971919	6.10989037717161	6.108280717431453
6	60.0	4.327311114762276	1	61.14791328780001	0	0.0000348	0	28.536529299199998	89.684442587	2.361571050429561	2.361536244952255
7	70.0	3.973852719130925	

array([( 0,   0.,  1.03960878, 0,            nan, nan,         nan, nan,             nan,           nan,         nan,         nan),
       ( 1,  10.,  1.04885742, 0,            nan, nan, 5.00000e-06,  0.,  2.73743775e+02,  157.61029242,  0.52388856,  0.52388359),
       ( 2,  20.,  1.93925115, 0, 8.46439718e+01,  0., 3.67500e-04,  0.,  1.39000625e+02, -136.35540326,  4.43901709,  4.43864961),
       ( 3,  30.,  2.18644167, 0, 9.16331731e+01,  0., 1.42400e-04,  0.,  4.46327400e+01,  136.26591312,  2.19244347,  2.1923011 ),
       ( 4,  40.,  2.49821775, 0, 7.86612036e+01,  0., 1.85000e-05,  0., -7.62721256e+01, -154.93332916,  1.26977721,  1.26975872),
       ( 5,  50.,  3.90977794, 0, 4.17617007e+00,  0., 1.60970e-03,  0.,  4.83359673e+01,   44.15979719,  6.10989038,  6.10828072),
       ( 6,  60.,  4.32731111, 1, 6.11479133e+01,  0., 3.48000e-05,  0.,  2.85365293e+01,   89.68444259,  2.36157105,  2.36153624),
       ( 7,  70.,  3.97385272, 0, 4.88187456e+01,  0., 3.83300e-04,  0.,  7.

In [10]:
## reproduce results for pair H in Figures 6a, 6c, 6e using the output above
## from the function cmp.spa_angpre_len_dif
pairH=cmp.ppf3('data/0.result_tables/pairH_LaurentiaRM_LaurentiaRMpo15degRe.txt')
#pairH_.txt without test
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc_nt(pairH,'0',0,100)
cmp.calc_nt(pairH,'1',100,200)
cmp.calc_nt(pairH,'2',200,300)
cmp.calc_nt(pairH,'3',300,400)
cmp.calc_nt(pairH,'4',400,530)
cmp.calc_nt(pairH,'5',0,530)

ds	da	dl	no	t0	t1
0.0601795	0.247907	1.09056e-05	0	0	100	0.102699
0.119489	0.228384	0.000109612	1	100	200	0.115994
0.189765	0.489795	2.70471e-05	2	200	300	0.226529
0.273107	0.357954	0.000110344	3	300	400	0.21039
0.291807	0.327833	0.000199642	4	400	530	0.206613
0.192786	0.328219	9.76309e-05	5	0	530	0.173701


In [11]:
## reproduce results for pair H in Figures 6b, 6d, 6f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairH.txt with test
print('ds\tda\tdl\tno\tt0\tt1')
cmp.calc(pairH,'0',0,100)
cmp.calc(pairH,'1',100,200)
cmp.calc(pairH,'2',200,300)
cmp.calc(pairH,'3',300,400)
cmp.calc(pairH,'4',400,530)
cmp.calc(pairH,'5',0,530)

ds	da	dl	no	t0	t1
0.00786784	0	0	0	0	100	0.00262261
0.0535331	0	0	1	100	200	0.0178444
0.189765	0.161511	0	2	200	300	0.117092
0.273107	0.0983773	2.24217e-05	3	300	400	0.123836
0.291807	0	7.21627e-07	4	400	530	0.0972694
0.170152	0.0449806	4.40752e-06	5	0	530	0.0717123


# The Above Procedure Applied on APWPs Without Interpolations

## pair A without interpolations

In [2]:
copy_tree("data/Laurentia_RM_NoIntp","/tmp/Laurentia_RM_NoIntp")
copy_tree("data/rot_version/Laurentia_RM_1deg_NoIntp","/tmp/Laurentia_RM_1deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_1deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairA_LaurentiaRM_LaurentiaRM1deg_NoIntp.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	0.06950468110321331	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	0.07012239182561895	0	nan	nan	0.	0	273.743775125	273.149982677	0.5238885648671224	0.5238885648601654
2	20.0	0.12965232767905435	0	0.	0	0.	0	139.00062497750002	139.0006249778	4.4390170905897985	4.439017090586511
3	30.0	0.14617923464677537	0	0.	0	0.	0	44.63274003309999	44.63274003229998	2.1924434711123775	2.1924434710977447
4	40.0	0.16702373999377892	0	0.	0	0.	0	-76.27212559900002	-76.272125598	1.2697772061265127	1.2697772061747379
5	50.0	0.2613763607482359	0	0.	0	0.	0	48.33596726459999	48.33596726339999	6.10989037717161	6.109890377109275
6	60.0	0.28930393757442735	0	0.	0	0.	0	28.536529299199998	28.53652929750001	2.361571050429561	2.3615710504797334
7	70.0	0.265656369681836	0	0.	0	0.	0	77.79573930800001	77.79573931	3.3378718721186993	3.337871872130717
8	80.0	0.27857

array([( 0,   0., 0.06950468, 0, nan, nan, nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10., 0.07012239, 0, nan, nan,  0.,  0.,  273.74377513,  273.14998268,  0.52388856,  0.52388856),
       ( 2,  20., 0.12965233, 0,  0.,  0.,  0.,  0.,  139.00062498,  139.00062498,  4.43901709,  4.43901709),
       ( 3,  30., 0.14617923, 0,  0.,  0.,  0.,  0.,   44.63274003,   44.63274003,  2.19244347,  2.19244347),
       ( 4,  40., 0.16702374, 0,  0.,  0.,  0.,  0.,  -76.2721256 ,  -76.2721256 ,  1.26977721,  1.26977721),
       ( 5,  50., 0.26137636, 0,  0.,  0.,  0.,  0.,   48.33596726,   48.33596726,  6.10989038,  6.10989038),
       ( 6,  60., 0.28930394, 0,  0.,  0.,  0.,  0.,   28.5365293 ,   28.5365293 ,  2.36157105,  2.36157105),
       ( 7,  70., 0.26565637, 0,  0.,  0.,  0.,  0.,   77.79573931,   77.79573931,  3.33787187,  3.33787187),
       ( 8,  80., 0.27857033, 0,  0.,  0.,  0.,  0.,  -38.12131862,  -38.12131862,  2.05979534,  2.05979534),
       ( 9

In [4]:
## reproduce results for pair A (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairAni=cmp.ppf3('data/0.result_tables/pairA_LaurentiaRM_LaurentiaRM1deg_NoIntp.txt')
#pairAni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairAni,'0',0,100)
cmp.calc_nt(pairAni,'1',100,200)
cmp.calc_nt(pairAni,'2',200,300)
cmp.calc_nt(pairAni,'3',300,400)
cmp.calc_nt(pairAni,'4',400,530)
cmp.calc_nt(pairAni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.00402313	0	0	0	0	100	0.00134104
0.00798601	0	0	1	100	200	0.002662
0.0126793	0	0	2	200	300	0.00422643
0.0178421	0	0	3	300	400	0.00594735
0.0195182	0	0	4	400	530	0.00650607
0.0119475	0	0	5	0	530	0.00398249


In [5]:
## reproduce results for pair A (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairAni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairAni,'0',0,100)
cmp.calc(pairAni,'1',100,200)
cmp.calc(pairAni,'2',200,300)
cmp.calc(pairAni,'3',300,400)
cmp.calc(pairAni,'4',400,530)
cmp.calc(pairAni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0	0	0	0	0	100	0
0	0	0	1	100	200	0
0	0	0	2	200	300	0
0	0	0	3	300	400	0
0	0	0	4	400	530	0
0	0	0	5	0	530	0


## pair B without interpolations

In [ ]:
copy_tree("data/rot_version/Laurentia_RM_15deg_NoIntp","/tmp/Laurentia_RM_15deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_15deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairB_LaurentiaRM_LaurentiaRM15deg_NoIntp.txt

In [18]:
## reproduce results for pair B (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairBni=cmp.ppf3('data/0.result_tables/pairB_LaurentiaRM_LaurentiaRM15deg_NoIntp.txt')
#pairBni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairBni,'0',0,100)
cmp.calc_nt(pairBni,'1',100,200)
cmp.calc_nt(pairBni,'2',200,300)
cmp.calc_nt(pairBni,'3',300,400)
cmp.calc_nt(pairBni,'4',400,530)
cmp.calc_nt(pairBni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0601857	0	0	0	0	100	0.0200619
0.119511	0	0	1	100	200	0.0398369
0.189876	0	0	2	200	300	0.0632921
0.267487	0	0	3	300	400	0.0891625
0.292735	0	0	4	400	530	0.0975782
0.179022	0	0	5	0	530	0.0596741


In [19]:
## reproduce results for pair B (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairBni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairBni,'0',0,100)
cmp.calc(pairBni,'1',100,200)
cmp.calc(pairBni,'2',200,300)
cmp.calc(pairBni,'3',300,400)
cmp.calc(pairBni,'4',400,530)
cmp.calc(pairBni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0	0	0	0	0	100	0
0.0245221	0	0	1	100	200	0.00817403
0.17214	0	0	2	200	300	0.0573801
0.267487	0	0	3	300	400	0.0891625
0.15983	0	0	4	400	530	0.0532768
0.109124	0	0	5	0	530	0.0363746


## pair C without interpolations

In [6]:
copy_tree("data/rot_version/Laurentia_RM_po_15deg_NoIntp","/tmp/Laurentia_RM_po_15deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_po_15deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairC_LaurentiaRM_LaurentiaRMpo15deg_NoIntp.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	1.0396087803797036	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.0488574183293418	0	nan	nan	0.000005	0	273.743775125	157.610292419	0.5238885648671224	0.523883588268831
2	20.0	1.939251148514321	0	84.64397175799996	0	0.0003675	0	139.00062497750002	-136.3554032645	4.4390170905897985	4.438649608107905
3	30.0	2.186441670519566	0	91.63317308890001	0	0.0001424	0	44.63274003309999	136.265913122	2.1924434711123775	2.192301096824105
4	40.0	2.4982177538741293	0	78.66120356299999	0	0.0000185	0	-76.27212559900002	-154.933329162	1.2697772061265127	1.2697587232430259
5	50.0	3.9097779362449967	0	4.176170072699989	0	0.0016097	0	48.33596726459999	44.1597971919	6.10989037717161	6.108280717431453
6	60.0	4.327311114762276	0	61.14791328780001	0	0.0000348	0	28.536529299199998	89.684442587	2.361571050429561	2.361536244952255
7	70.0	3.973852719130925	

array([( 0,   0.,  1.03960878, 0,            nan, nan,            nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  1.04885742, 0,            nan, nan, 5.00000000e-06,  0.,  273.74377513,  157.61029242,  0.52388856,  0.52388359),
       ( 2,  20.,  1.93925115, 0, 8.46439718e+01,  0., 3.67500000e-04,  0.,  139.00062498, -136.35540326,  4.43901709,  4.43864961),
       ( 3,  30.,  2.18644167, 0, 9.16331731e+01,  0., 1.42400000e-04,  0.,   44.63274003,  136.26591312,  2.19244347,  2.1923011 ),
       ( 4,  40.,  2.49821775, 0, 7.86612036e+01,  0., 1.85000000e-05,  0.,  -76.2721256 , -154.93332916,  1.26977721,  1.26975872),
       ( 5,  50.,  3.90977794, 0, 4.17617007e+00,  0., 1.60970000e-03,  0.,   48.33596726,   44.15979719,  6.10989038,  6.10828072),
       ( 6,  60.,  4.32731111, 0, 6.11479133e+01,  0., 3.48000000e-05,  0.,   28.5365293 ,   89.68444259,  2.36157105,  2.36153624),
       ( 7,  70.,  3.97385272, 0, 4.88187456e+01,  0., 3.83300000e-04

In [9]:
## reproduce results for pair C (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairCni=cmp.ppf3('data/0.result_tables/pairC_LaurentiaRM_LaurentiaRMpo15deg_NoIntp.txt')
#pairCni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairCni,'0',0,100)
cmp.calc_nt(pairCni,'1',100,200)
cmp.calc_nt(pairCni,'2',200,300)
cmp.calc_nt(pairCni,'3',300,400)
cmp.calc_nt(pairCni,'4',400,530)
cmp.calc_nt(pairCni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0601795	0.247907	1.09056e-05	0	0	100	0.102699
0.119489	0.228384	0.000109612	1	100	200	0.115994
0.189765	0.489795	2.70471e-05	2	200	300	0.226529
0.267239	0.53888	0.0150239	3	300	400	0.273714
0.292594	0.249715	0.0160828	4	400	530	0.186131
0.178939	0.330728	0.00680738	5	0	530	0.172158


In [10]:
## reproduce results for pair C (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairCni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairCni,'0',0,100)
cmp.calc(pairCni,'1',100,200)
cmp.calc(pairCni,'2',200,300)
cmp.calc(pairCni,'3',300,400)
cmp.calc(pairCni,'4',400,530)
cmp.calc(pairCni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0	0	0	0	0	100	0
0.0245112	0	0	1	100	200	0.00817039
0.172042	0.101838	0	2	200	300	0.0912932
0.267239	0	0	3	300	400	0.0890797
0.159721	0	0	4	400	530	0.0532402
0.109055	0.0203676	0	5	0	530	0.0431408


## pair D without interpolations

In [7]:
copy_tree("data/rot_version/Laurentia_RM_p_15deg_NoIntp","/tmp/Laurentia_RM_p_15deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_p_15deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairD_LaurentiaRM_LaurentiaRMp15deg_NoIntp.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	1.0396227823854567	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.0488575638182625	0	nan	nan	1.9611886168159605	0	273.743775125	223.83279393	0.5238885648671224	2.485077181683083
2	20.0	1.9393151442350478	0	10.996304058599975	0	1.0576883730015094	0	139.00062497750002	149.9969290361	4.4390170905897985	5.496705463591308
3	30.0	2.1864678002853206	0	38.351504983900014	0	3.9440477103645275	0	44.63274003309999	82.984245017	2.1924434711123775	6.136491181476905
4	40.0	2.498240508742987	0	136.48827074899998	0	0.868194925161442	0	-76.27212559900002	147.239603652	1.2697772061265127	2.1379721312879547
5	50.0	3.9096775901328615	0	125.94344157170002	0	3.8611309391005415	0	48.33596726459999	174.2794088363	6.10989037717161	9.971021316272152
6	60.0	4.3280044010637	0	141.8874984952	0	3.9137624281754877	0	28.536529299199998	-113.350969196	2.361571

array([( 0,   0.,  1.03962278, 0,          nan, nan,         nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  1.04885756, 0,          nan, nan,  1.96118862,  0.,  273.74377513,  223.83279393,  0.52388856,  2.48507718),
       ( 2,  20.,  1.93931514, 0,  10.99630406,  0.,  1.05768837,  0.,  139.00062498,  149.99692904,  4.43901709,  5.49670546),
       ( 3,  30.,  2.1864678 , 0,  38.35150498,  0.,  3.94404771,  0.,   44.63274003,   82.98424502,  2.19244347,  6.13649118),
       ( 4,  40.,  2.49824051, 0, 136.48827075,  0.,  0.86819493,  0.,  -76.2721256 ,  147.23960365,  1.26977721,  2.13797213),
       ( 5,  50.,  3.90967759, 0, 125.94344157,  0.,  3.86113094,  0.,   48.33596726,  174.27940884,  6.10989038,  9.97102132),
       ( 6,  60.,  4.3280044 , 0, 141.8874985 ,  0.,  3.91376243,  0.,   28.5365293 , -113.3509692 ,  2.36157105,  6.27533348),
       ( 7,  70.,  3.9735243 , 0, 146.84572442,  0.,  1.7565044 ,  0.,   77.79573931, -135.35853627,  3.

In [11]:
## reproduce results for pair D (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairDni=cmp.ppf3('data/0.result_tables/pairD_LaurentiaRM_LaurentiaRMp15deg_NoIntp.txt')
#pairDni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairDni,'0',0,100)
cmp.calc_nt(pairDni,'1',100,200)
cmp.calc_nt(pairDni,'2',200,300)
cmp.calc_nt(pairDni,'3',300,400)
cmp.calc_nt(pairDni,'4',400,530)
cmp.calc_nt(pairDni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0601805	0.50209	0.124247	0	0	100	0.228839
0.119469	0.592204	0.132934	1	100	200	0.281536
0.189762	0.580493	0.366091	2	200	300	0.378782
0.267308	0.432998	0.219157	3	300	400	0.306488
0.292416	0.476273	0.290317	4	400	530	0.353002
0.178896	0.51743	0.230158	5	0	530	0.308828


In [12]:
## reproduce results for pair D (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairDni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairDni,'0',0,100)
cmp.calc(pairDni,'1',100,200)
cmp.calc(pairDni,'2',200,300)
cmp.calc(pairDni,'3',300,400)
cmp.calc(pairDni,'4',400,530)
cmp.calc(pairDni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0	0	0	0	0	100	0
0.0245146	0	0	1	100	200	0.00817154
0.172032	0.0975317	0.169474	2	200	300	0.146346
0.267308	0.116095	0.0549914	3	300	400	0.146132
0.15968	0	0.0756355	4	400	530	0.0784385
0.109048	0.0324058	0.0609041	5	0	530	0.0674525


## pair E without interpolations

In [ ]:
copy_tree("data/rot_version/Laurentia_RM_45deg_NoIntp","/tmp/Laurentia_RM_45deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_45deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairE_LaurentiaRM_LaurentiaRM45deg_NoIntp.txt

In [3]:
## reproduce results for pair E (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairEni=cmp.ppf3('data/0.result_tables/pairE_LaurentiaRM_LaurentiaRM45deg_NoIntp.txt')
#pairEni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairEni,'0',0,100)
cmp.calc_nt(pairEni,'1',100,200)
cmp.calc_nt(pairEni,'2',200,300)
cmp.calc_nt(pairEni,'3',300,400)
cmp.calc_nt(pairEni,'4',400,530)
cmp.calc_nt(pairEni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.176683	0	0	0	0	100	0.0588942
0.351769	0	0	1	100	200	0.117256
0.56193	0	0	2	200	300	0.18731
0.79885	0	0	3	300	400	0.266283
0.87722	0	0	4	400	530	0.292407
0.532422	0	0	5	0	530	0.177474


In [4]:
## reproduce results for pair E (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairEni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairEni,'0',0,100)
cmp.calc(pairEni,'1',100,200)
cmp.calc(pairEni,'2',200,300)
cmp.calc(pairEni,'3',300,400)
cmp.calc(pairEni,'4',400,530)
cmp.calc(pairEni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.134189	0	0	0	0	100	0.0447297
0.330752	0	0	1	100	200	0.110251
0.56193	0	0	2	200	300	0.18731
0.79885	0	0	3	300	400	0.266283
0.87722	0	0	4	400	530	0.292407
0.522476	0	0	5	0	530	0.174159


## pair F without interpolations

In [5]:
copy_tree("data/Laurentia_RM_NoErr_NoIntp","/tmp/Laurentia_RM_NoErr_NoIntp")
copy_tree("data/rot_version/Laurentia_RM_NoErr_45deg_NoIntp","/tmp/Laurentia_RM_NoErr_45deg_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoErr_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_NoErr_45deg_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairF_LaurentiaRMnoErr_LaurentiaRMnoErr45deg_NoIntp.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	3.048337524892148	1	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	3.075435566736447	1	nan	nan	0.	0	273.743775125	251.624883576	0.5238885648671224	0.5238885648866021
2	20.0	5.6879579579765975	1	0.	0	0.	0	139.00062497750002	139.0006249855	4.4390170905897985	4.439017090566704
3	30.0	6.413722136661157	1	0.	0	0.	0	44.63274003309999	44.632740032399994	2.1924434711123775	2.1924434711005714
4	40.0	7.329460598076736	1	0.	0	0.	0	-76.27212559900002	-76.27212559899999	1.2697772061265127	1.2697772061365598
5	50.0	11.481287633763639	1	0.	0	0.	0	48.33596726459999	48.33596726320002	6.10989037717161	6.109890377219841
6	60.0	12.712849877468846	1	0.	0	0.	0	28.536529299199998	28.536529299899996	2.361571050429561	2.3615710504073313
7	70.0	11.669939449301262	1	0.	0	0.	0	77.79573930800001	77.79573930800001	3.3378718721186993	3.337871872117279
8	80.0	12.

array([( 0,   0.,  3.04833752, 1, nan, nan, nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  3.07543557, 1, nan, nan,  0.,  0.,  273.74377513,  251.62488358,  0.52388856,  0.52388856),
       ( 2,  20.,  5.68795796, 1,  0.,  0.,  0.,  0.,  139.00062498,  139.00062499,  4.43901709,  4.43901709),
       ( 3,  30.,  6.41372214, 1,  0.,  0.,  0.,  0.,   44.63274003,   44.63274003,  2.19244347,  2.19244347),
       ( 4,  40.,  7.3294606 , 1,  0.,  0.,  0.,  0.,  -76.2721256 ,  -76.2721256 ,  1.26977721,  1.26977721),
       ( 5,  50., 11.48128763, 1,  0.,  0.,  0.,  0.,   48.33596726,   48.33596726,  6.10989038,  6.10989038),
       ( 6,  60., 12.71284988, 1,  0.,  0.,  0.,  0.,   28.5365293 ,   28.5365293 ,  2.36157105,  2.36157105),
       ( 7,  70., 11.66993945, 1,  0.,  0.,  0.,  0.,   77.79573931,   77.79573931,  3.33787187,  3.33787187),
       ( 8,  80., 12.23934811, 1,  0.,  0.,  0.,  0.,  -38.12131862,  -38.12131862,  2.05979534,  2.05979534),
 

In [15]:
## reproduce results for pair F (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairFni=cmp.ppf3('data/0.result_tables/pairF_LaurentiaRMnoErr_LaurentiaRMnoErr45deg_NoIntp.txt')
#pairFni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairFni,'0',0,100)
cmp.calc_nt(pairFni,'1',100,200)
cmp.calc_nt(pairFni,'2',200,300)
cmp.calc_nt(pairFni,'3',300,400)
cmp.calc_nt(pairFni,'4',400,530)
cmp.calc_nt(pairFni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.176683	0	0	0	0	100	0.0588942
0.351769	0	0	1	100	200	0.117256
0.56193	0	0	2	200	300	0.18731
0.79885	0	0	3	300	400	0.266283
0.87722	0	0	4	400	530	0.292407
0.532422	0	0	5	0	530	0.177474


In [16]:
## reproduce results for pair F (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairFni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairFni,'0',0,100)
cmp.calc(pairFni,'1',100,200)
cmp.calc(pairFni,'2',200,300)
cmp.calc(pairFni,'3',300,400)
cmp.calc(pairFni,'4',400,530)
cmp.calc(pairFni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.176683	0	0	0	0	100	0.0588942
0.351769	0	0	1	100	200	0.117256
0.56193	0	0	2	200	300	0.18731
0.79885	0	0	3	300	400	0.266283
0.87722	0	0	4	400	530	0.292407
0.532422	0	0	5	0	530	0.177474


## pair H without interpolations

In [8]:
copy_tree("data/Laurentia_RM_NoIntp","/tmp/Laurentia_RM_NoIntp")
cmp.spa_angpre_len_dif('data/Laurentia_RM_NoIntp.txt',
                       'data/rot_version/Laurentia_RM_po_15deg_re_NoIntp.txt',
                       pnh1=1,pnh2=0)
# the results should be the same as stored in
# data/0.result_tables/pairH_LaurentiaRM_LaurentiaRMpo15degRe_NoIntp.txt

00_no	01_tstop	10_spa_pol_dif	11_spa_pol_tes	20_ang_seg_dif	21_ang_seg_tes	30_len_seg_dif	31_len_seg_tes	22_course_seg1	23_course_seg2	32_len_seg1	33_len_seg2
0	0.0	1.0396087803797036	0	nan	nan	nan	nan	nan	nan	nan	nan
1	10.0	1.0488574183293418	0	nan	nan	0.000005	0	273.743775125	157.610292419	0.5238885648671224	0.523883588268831
2	20.0	1.939251148514321	0	84.64397175799996	0	0.0003675	0	139.00062497750002	-136.3554032645	4.4390170905897985	4.438649608107905
3	30.0	2.186441670519566	0	91.63317308890001	0	0.0001424	0	44.63274003309999	136.265913122	2.1924434711123775	2.192301096824105
4	40.0	2.4982177538741293	0	78.66120356299999	0	0.0000185	0	-76.27212559900002	-154.933329162	1.2697772061265127	1.2697587232430259
5	50.0	3.9097779362449967	0	4.176170072699989	0	0.0016097	0	48.33596726459999	44.1597971919	6.10989037717161	6.108280717431453
6	60.0	4.327311114762276	1	61.14791328780001	0	0.0000348	0	28.536529299199998	89.684442587	2.361571050429561	2.361536244952255
7	70.0	3.973852719130925	

array([( 0,   0.,  1.03960878, 0,            nan, nan,            nan, nan,           nan,           nan,         nan,         nan),
       ( 1,  10.,  1.04885742, 0,            nan, nan, 5.00000000e-06,  0.,  273.74377513,  157.61029242,  0.52388856,  0.52388359),
       ( 2,  20.,  1.93925115, 0, 8.46439718e+01,  0., 3.67500000e-04,  0.,  139.00062498, -136.35540326,  4.43901709,  4.43864961),
       ( 3,  30.,  2.18644167, 0, 9.16331731e+01,  0., 1.42400000e-04,  0.,   44.63274003,  136.26591312,  2.19244347,  2.1923011 ),
       ( 4,  40.,  2.49821775, 0, 7.86612036e+01,  0., 1.85000000e-05,  0.,  -76.2721256 , -154.93332916,  1.26977721,  1.26975872),
       ( 5,  50.,  3.90977794, 0, 4.17617007e+00,  0., 1.60970000e-03,  0.,   48.33596726,   44.15979719,  6.10989038,  6.10828072),
       ( 6,  60.,  4.32731111, 1, 6.11479133e+01,  0., 3.48000000e-05,  0.,   28.5365293 ,   89.68444259,  2.36157105,  2.36153624),
       ( 7,  70.,  3.97385272, 0, 4.88187456e+01,  0., 3.83300000e-04

In [12]:
## reproduce results for pair H (without interpolations) in Figures 7a, 7c, 7e using the output above
## from the function cmp.spa_angpre_len_dif
pairHni=cmp.ppf3('data/0.result_tables/pairH_LaurentiaRM_LaurentiaRMpo15degRe_NoIntp.txt')
#pairHni_.txt without test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc_nt(pairHni,'0',0,100)
cmp.calc_nt(pairHni,'1',100,200)
cmp.calc_nt(pairHni,'2',200,300)
cmp.calc_nt(pairHni,'3',300,400)
cmp.calc_nt(pairHni,'4',400,530)
cmp.calc_nt(pairHni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.0601795	0.247907	1.09056e-05	0	0	100	0.102699
0.119489	0.228384	0.000109612	1	100	200	0.115994
0.189765	0.489795	2.70471e-05	2	200	300	0.226529
0.267239	0.53888	0.0150239	3	300	400	0.273714
0.292594	0.249715	0.0160828	4	400	530	0.186131
0.178939	0.330728	0.00680738	5	0	530	0.172158


In [13]:
## reproduce results for pair H (without interpolations) in Figures 7b, 7d, 7f using the same output above
## from the function cmp.spa_angpre_len_dif
#pairHni.txt with test
print('ds\tda\tdl\tno\tt0\tt1\tequalWeighCPD')
cmp.calc(pairHni,'0',0,100)
cmp.calc(pairHni,'1',100,200)
cmp.calc(pairHni,'2',200,300)
cmp.calc(pairHni,'3',300,400)
cmp.calc(pairHni,'4',400,530)
cmp.calc(pairHni,'5',0,530)

ds	da	dl	no	t0	t1	equalWeighCPD
0.00786784	0	0	0	0	100	0.00262261
0.0535331	0	0	1	100	200	0.0178444
0.189765	0.161511	0	2	200	300	0.117092
0.267239	0.131389	0	3	300	400	0.132876
0.292594	0	0	4	400	530	0.0975313
0.152935	0.0469009	0	5	0	530	0.0666119
